// akeyless/dist/model/AccessOrGroupPermissionAssignment.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AccessOrGroupPermissionAssignment model module.
 * @module model/AccessOrGroupPermissionAssignment
 * @version 3.5.0
 */
var AccessOrGroupPermissionAssignment = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AccessOrGroupPermissionAssignment</code>.
   * @alias module:model/AccessOrGroupPermissionAssignment
   */
  function AccessOrGroupPermissionAssignment() {
    _classCallCheck(this, AccessOrGroupPermissionAssignment);

    AccessOrGroupPermissionAssignment.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AccessOrGroupPermissionAssignment, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AccessOrGroupPermissionAssignment</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AccessOrGroupPermissionAssignment} obj Optional instance to populate.
     * @return {module:model/AccessOrGroupPermissionAssignment} The populated <code>AccessOrGroupPermissionAssignment</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AccessOrGroupPermissionAssignment();

        if (data.hasOwnProperty('access_id')) {
          obj['access_id'] = _ApiClient["default"].convertToType(data['access_id'], 'String');
        }

        if (data.hasOwnProperty('access_type')) {
          obj['access_type'] = _ApiClient["default"].convertToType(data['access_type'], 'String');
        }

        if (data.hasOwnProperty('assignment_name')) {
          obj['assignment_name'] = _ApiClient["default"].convertToType(data['assignment_name'], 'String');
        }

        if (data.hasOwnProperty('assignment_type')) {
          obj['assignment_type'] = _ApiClient["default"].convertToType(data['assignment_type'], 'String');
        }

        if (data.hasOwnProperty('group_id')) {
          obj['group_id'] = _ApiClient["default"].convertToType(data['group_id'], 'String');
        }

        if (data.hasOwnProperty('sub_claims')) {
          obj['sub_claims'] = _ApiClient["default"].convertToType(data['sub_claims'], {
            'String': ['String']
          });
        }
      }

      return obj;
    }
  }]);

  return AccessOrGroupPermissionAssignment;
}();
/**
 * @member {String} access_id
 */


AccessOrGroupPermissionAssignment.prototype['access_id'] = undefined;
/**
 * @member {String} access_type
 */

AccessOrGroupPermissionAssignment.prototype['access_type'] = undefined;
/**
 * @member {String} assignment_name
 */

AccessOrGroupPermissionAssignment.prototype['assignment_name'] = undefined;
/**
 * @member {String} assignment_type
 */

AccessOrGroupPermissionAssignment.prototype['assignment_type'] = undefined;
/**
 * @member {String} group_id
 */

AccessOrGroupPermissionAssignment.prototype['group_id'] = undefined;
/**
 * @member {Object.<String, Array.<String>>} sub_claims
 */

AccessOrGroupPermissionAssignment.prototype['sub_claims'] = undefined;
var _default = AccessOrGroupPermissionAssignment;
exports["default"] = _default;
