// akeyless/dist/model/AccountGeneralSettings.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DataProtectionSection = _interopRequireDefault(require("./DataProtectionSection"));

var _PasswordPolicyInfo = _interopRequireDefault(require("./PasswordPolicyInfo"));

var _SharingPolicyInfo = _interopRequireDefault(require("./SharingPolicyInfo"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AccountGeneralSettings model module.
 * @module model/AccountGeneralSettings
 * @version 3.3.9
 */
var AccountGeneralSettings = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AccountGeneralSettings</code>.
   * AccountGeneralSettings describes general settings for an account
   * @alias module:model/AccountGeneralSettings
   */
  function AccountGeneralSettings() {
    _classCallCheck(this, AccountGeneralSettings);

    AccountGeneralSettings.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AccountGeneralSettings, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AccountGeneralSettings</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AccountGeneralSettings} obj Optional instance to populate.
     * @return {module:model/AccountGeneralSettings} The populated <code>AccountGeneralSettings</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AccountGeneralSettings();

        if (data.hasOwnProperty('data_protection_section')) {
          obj['data_protection_section'] = _DataProtectionSection["default"].constructFromObject(data['data_protection_section']);
        }

        if (data.hasOwnProperty('enable_request_for_access')) {
          obj['enable_request_for_access'] = _ApiClient["default"].convertToType(data['enable_request_for_access'], 'Boolean');
        }

        if (data.hasOwnProperty('password_policy')) {
          obj['password_policy'] = _PasswordPolicyInfo["default"].constructFromObject(data['password_policy']);
        }

        if (data.hasOwnProperty('protect_items_by_default')) {
          obj['protect_items_by_default'] = _ApiClient["default"].convertToType(data['protect_items_by_default'], 'Boolean');
        }

        if (data.hasOwnProperty('sharing_policy')) {
          obj['sharing_policy'] = _SharingPolicyInfo["default"].constructFromObject(data['sharing_policy']);
        }
      }

      return obj;
    }
  }]);

  return AccountGeneralSettings;
}();
/**
 * @member {module:model/DataProtectionSection} data_protection_section
 */


AccountGeneralSettings.prototype['data_protection_section'] = undefined;
/**
 * @member {Boolean} enable_request_for_access
 */

AccountGeneralSettings.prototype['enable_request_for_access'] = undefined;
/**
 * @member {module:model/PasswordPolicyInfo} password_policy
 */

AccountGeneralSettings.prototype['password_policy'] = undefined;
/**
 * @member {Boolean} protect_items_by_default
 */

AccountGeneralSettings.prototype['protect_items_by_default'] = undefined;
/**
 * @member {module:model/SharingPolicyInfo} sharing_policy
 */

AccountGeneralSettings.prototype['sharing_policy'] = undefined;
var _default = AccountGeneralSettings;
exports["default"] = _default;
