// akeyless/dist/model/AllowedAccess.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AllowedAccess model module.
 * @module model/AllowedAccess
 * @version 3.5.0
 */
var AllowedAccess = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AllowedAccess</code>.
   * @alias module:model/AllowedAccess
   */
  function AllowedAccess() {
    _classCallCheck(this, AllowedAccess);

    AllowedAccess.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AllowedAccess, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AllowedAccess</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AllowedAccess} obj Optional instance to populate.
     * @return {module:model/AllowedAccess} The populated <code>AllowedAccess</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AllowedAccess();

        if (data.hasOwnProperty('access_id')) {
          obj['access_id'] = _ApiClient["default"].convertToType(data['access_id'], 'String');
        }

        if (data.hasOwnProperty('access_type')) {
          obj['access_type'] = _ApiClient["default"].convertToType(data['access_type'], 'String');
        }

        if (data.hasOwnProperty('cluster_id')) {
          obj['cluster_id'] = _ApiClient["default"].convertToType(data['cluster_id'], 'Number');
        }

        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('editable')) {
          obj['editable'] = _ApiClient["default"].convertToType(data['editable'], 'Boolean');
        }

        if (data.hasOwnProperty('error')) {
          obj['error'] = _ApiClient["default"].convertToType(data['error'], 'String');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }

        if (data.hasOwnProperty('is_valid')) {
          obj['is_valid'] = _ApiClient["default"].convertToType(data['is_valid'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('permissions')) {
          obj['permissions'] = _ApiClient["default"].convertToType(data['permissions'], ['String']);
        }

        if (data.hasOwnProperty('sub_claims')) {
          obj['sub_claims'] = _ApiClient["default"].convertToType(data['sub_claims'], {
            'String': ['String']
          });
        }

        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
      }

      return obj;
    }
  }]);

  return AllowedAccess;
}();
/**
 * @member {String} access_id
 */


AllowedAccess.prototype['access_id'] = undefined;
/**
 * @member {String} access_type
 */

AllowedAccess.prototype['access_type'] = undefined;
/**
 * @member {Number} cluster_id
 */

AllowedAccess.prototype['cluster_id'] = undefined;
/**
 * @member {Date} created_at
 */

AllowedAccess.prototype['created_at'] = undefined;
/**
 * @member {String} description
 */

AllowedAccess.prototype['description'] = undefined;
/**
 * @member {Boolean} editable
 */

AllowedAccess.prototype['editable'] = undefined;
/**
 * @member {String} error
 */

AllowedAccess.prototype['error'] = undefined;
/**
 * @member {Number} id
 */

AllowedAccess.prototype['id'] = undefined;
/**
 * @member {Boolean} is_valid
 */

AllowedAccess.prototype['is_valid'] = undefined;
/**
 * @member {String} name
 */

AllowedAccess.prototype['name'] = undefined;
/**
 * @member {Array.<String>} permissions
 */

AllowedAccess.prototype['permissions'] = undefined;
/**
 * @member {Object.<String, Array.<String>>} sub_claims
 */

AllowedAccess.prototype['sub_claims'] = undefined;
/**
 * @member {Date} updated_at
 */

AllowedAccess.prototype['updated_at'] = undefined;
var _default = AllowedAccess;
exports["default"] = _default;
