// akeyless/dist/model/AllowedAccessUpdateArgs.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AllowedAccessUpdateArgs model module.
 * @module model/AllowedAccessUpdateArgs
 * @version 3.3.5
 */
var AllowedAccessUpdateArgs = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AllowedAccessUpdateArgs</code>.
   * @alias module:model/AllowedAccessUpdateArgs
   * @param accessId {String} Access ID The access id to be attached to this allowed access. Auth method with this access id should already exist.
   * @param name {String} Allowed access name
   */
  function AllowedAccessUpdateArgs(accessId, name) {
    _classCallCheck(this, AllowedAccessUpdateArgs);

    AllowedAccessUpdateArgs.initialize(this, accessId, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AllowedAccessUpdateArgs, null, [{
    key: "initialize",
    value: function initialize(obj, accessId, name) {
      obj['access-id'] = accessId;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>AllowedAccessUpdateArgs</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AllowedAccessUpdateArgs} obj Optional instance to populate.
     * @return {module:model/AllowedAccessUpdateArgs} The populated <code>AllowedAccessUpdateArgs</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AllowedAccessUpdateArgs();

        if (data.hasOwnProperty('access-id')) {
          obj['access-id'] = _ApiClient["default"].convertToType(data['access-id'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('permissions')) {
          obj['permissions'] = _ApiClient["default"].convertToType(data['permissions'], 'String');
        }

        if (data.hasOwnProperty('sub-claims')) {
          obj['sub-claims'] = _ApiClient["default"].convertToType(data['sub-claims'], {
            'String': 'String'
          });
        }
      }

      return obj;
    }
  }]);

  return AllowedAccessUpdateArgs;
}();
/**
 * Access ID The access id to be attached to this allowed access. Auth method with this access id should already exist.
 * @member {String} access-id
 */


AllowedAccessUpdateArgs.prototype['access-id'] = undefined;
/**
 * Allowed access description
 * @member {String} description
 */

AllowedAccessUpdateArgs.prototype['description'] = undefined;
/**
 * Allowed access name
 * @member {String} name
 */

AllowedAccessUpdateArgs.prototype['name'] = undefined;
/**
 * New allowed access name
 * @member {String} new-name
 */

AllowedAccessUpdateArgs.prototype['new-name'] = undefined;
/**
 * Permissions  Comma-seperated list of permissions for this allowed access. Available permissions: [defaults,targets,classic_keys,automatic_migration,ldap_auth,dynamic_secret,k8s_auth,log_forwarding,zero_knowledge_encryption,rotated_secret,caching,event_forwarding,admin,kmip,general]
 * @member {String} permissions
 */

AllowedAccessUpdateArgs.prototype['permissions'] = undefined;
/**
 * Sub claims key/val of sub claims, e.g group=admins,developers
 * @member {Object.<String, String>} sub-claims
 */

AllowedAccessUpdateArgs.prototype['sub-claims'] = undefined;
var _default = AllowedAccessUpdateArgs;
exports["default"] = _default;
