// akeyless/dist/model/AllowedAccessOld.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AllowedAccessOld model module.
 * @module model/AllowedAccessOld
 * @version 3.5.0
 */
var AllowedAccessOld = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AllowedAccessOld</code>.
   * Deprecated: AllowedAccessOld please use Gator allowed_access API structs such as AllowedAccessInput/AllowedAccess
   * @alias module:model/AllowedAccessOld
   */
  function AllowedAccessOld() {
    _classCallCheck(this, AllowedAccessOld);

    AllowedAccessOld.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AllowedAccessOld, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AllowedAccessOld</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AllowedAccessOld} obj Optional instance to populate.
     * @return {module:model/AllowedAccessOld} The populated <code>AllowedAccessOld</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AllowedAccessOld();

        if (data.hasOwnProperty('acc_id')) {
          obj['acc_id'] = _ApiClient["default"].convertToType(data['acc_id'], 'String');
        }

        if (data.hasOwnProperty('access_permissions')) {
          obj['access_permissions'] = _ApiClient["default"].convertToType(data['access_permissions'], ['String']);
        }

        if (data.hasOwnProperty('access_rules_type')) {
          obj['access_rules_type'] = _ApiClient["default"].convertToType(data['access_rules_type'], 'String');
        }

        if (data.hasOwnProperty('allowed_api')) {
          obj['allowed_api'] = _ApiClient["default"].convertToType(data['allowed_api'], 'Boolean');
        }

        if (data.hasOwnProperty('alloweds_login')) {
          obj['alloweds_login'] = _ApiClient["default"].convertToType(data['alloweds_login'], 'Boolean');
        }

        if (data.hasOwnProperty('editable')) {
          obj['editable'] = _ApiClient["default"].convertToType(data['editable'], 'Boolean');
        }

        if (data.hasOwnProperty('err_msg')) {
          obj['err_msg'] = _ApiClient["default"].convertToType(data['err_msg'], 'String');
        }

        if (data.hasOwnProperty('hash')) {
          obj['hash'] = _ApiClient["default"].convertToType(data['hash'], 'String');
        }

        if (data.hasOwnProperty('is_valid')) {
          obj['is_valid'] = _ApiClient["default"].convertToType(data['is_valid'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('sub_claims')) {
          obj['sub_claims'] = _ApiClient["default"].convertToType(data['sub_claims'], {
            'String': ['String']
          });
        }
      }

      return obj;
    }
  }]);

  return AllowedAccessOld;
}();
/**
 * @member {String} acc_id
 */


AllowedAccessOld.prototype['acc_id'] = undefined;
/**
 * @member {Array.<String>} access_permissions
 */

AllowedAccessOld.prototype['access_permissions'] = undefined;
/**
 * @member {String} access_rules_type
 */

AllowedAccessOld.prototype['access_rules_type'] = undefined;
/**
 * @member {Boolean} allowed_api
 */

AllowedAccessOld.prototype['allowed_api'] = undefined;
/**
 * @member {Boolean} alloweds_login
 */

AllowedAccessOld.prototype['alloweds_login'] = undefined;
/**
 * @member {Boolean} editable
 */

AllowedAccessOld.prototype['editable'] = undefined;
/**
 * @member {String} err_msg
 */

AllowedAccessOld.prototype['err_msg'] = undefined;
/**
 * @member {String} hash
 */

AllowedAccessOld.prototype['hash'] = undefined;
/**
 * @member {Boolean} is_valid
 */

AllowedAccessOld.prototype['is_valid'] = undefined;
/**
 * @member {String} name
 */

AllowedAccessOld.prototype['name'] = undefined;
/**
 * @member {Object.<String, Array.<String>>} sub_claims
 */

AllowedAccessOld.prototype['sub_claims'] = undefined;
var _default = AllowedAccessOld;
exports["default"] = _default;
