// akeyless/dist/model/CertificateChainInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CertificateExpirationEvent = _interopRequireDefault(require("./CertificateExpirationEvent"));

var _CertificateInfo = _interopRequireDefault(require("./CertificateInfo"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CertificateChainInfo model module.
 * @module model/CertificateChainInfo
 * @version 3.3.14
 */
var CertificateChainInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CertificateChainInfo</code>.
   * @alias module:model/CertificateChainInfo
   */
  function CertificateChainInfo() {
    _classCallCheck(this, CertificateChainInfo);

    CertificateChainInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CertificateChainInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CertificateChainInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CertificateChainInfo} obj Optional instance to populate.
     * @return {module:model/CertificateChainInfo} The populated <code>CertificateChainInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CertificateChainInfo();

        if (data.hasOwnProperty('certificate_chain')) {
          obj['certificate_chain'] = _ApiClient["default"].convertToType(data['certificate_chain'], [_CertificateInfo["default"]]);
        }

        if (data.hasOwnProperty('certificate_format')) {
          obj['certificate_format'] = _ApiClient["default"].convertToType(data['certificate_format'], 'String');
        }

        if (data.hasOwnProperty('certificate_issuer_item_id')) {
          obj['certificate_issuer_item_id'] = _ApiClient["default"].convertToType(data['certificate_issuer_item_id'], 'Number');
        }

        if (data.hasOwnProperty('certificate_issuer_name')) {
          obj['certificate_issuer_name'] = _ApiClient["default"].convertToType(data['certificate_issuer_name'], 'String');
        }

        if (data.hasOwnProperty('certificate_pem')) {
          obj['certificate_pem'] = _ApiClient["default"].convertToType(data['certificate_pem'], 'String');
        }

        if (data.hasOwnProperty('certificate_status')) {
          obj['certificate_status'] = _ApiClient["default"].convertToType(data['certificate_status'], 'String');
        }

        if (data.hasOwnProperty('expiration_events')) {
          obj['expiration_events'] = _ApiClient["default"].convertToType(data['expiration_events'], [_CertificateExpirationEvent["default"]]);
        }
      }

      return obj;
    }
  }]);

  return CertificateChainInfo;
}();
/**
 * @member {Array.<module:model/CertificateInfo>} certificate_chain
 */


CertificateChainInfo.prototype['certificate_chain'] = undefined;
/**
 * @member {String} certificate_format
 */

CertificateChainInfo.prototype['certificate_format'] = undefined;
/**
 * @member {Number} certificate_issuer_item_id
 */

CertificateChainInfo.prototype['certificate_issuer_item_id'] = undefined;
/**
 * @member {String} certificate_issuer_name
 */

CertificateChainInfo.prototype['certificate_issuer_name'] = undefined;
/**
 * @member {String} certificate_pem
 */

CertificateChainInfo.prototype['certificate_pem'] = undefined;
/**
 * @member {String} certificate_status
 */

CertificateChainInfo.prototype['certificate_status'] = undefined;
/**
 * @member {Array.<module:model/CertificateExpirationEvent>} expiration_events
 */

CertificateChainInfo.prototype['expiration_events'] = undefined;
var _default = CertificateChainInfo;
exports["default"] = _default;
