// akeyless/dist/model/CertificateTemplateInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CertificateTemplateInfo model module.
 * @module model/CertificateTemplateInfo
 * @version 3.3.5
 */
var CertificateTemplateInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CertificateTemplateInfo</code>.
   * @alias module:model/CertificateTemplateInfo
   */
  function CertificateTemplateInfo() {
    _classCallCheck(this, CertificateTemplateInfo);

    CertificateTemplateInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CertificateTemplateInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CertificateTemplateInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CertificateTemplateInfo} obj Optional instance to populate.
     * @return {module:model/CertificateTemplateInfo} The populated <code>CertificateTemplateInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CertificateTemplateInfo();

        if (data.hasOwnProperty('common_name')) {
          obj['common_name'] = _ApiClient["default"].convertToType(data['common_name'], 'String');
        }

        if (data.hasOwnProperty('country')) {
          obj['country'] = _ApiClient["default"].convertToType(data['country'], 'String');
        }

        if (data.hasOwnProperty('digest_algo')) {
          obj['digest_algo'] = _ApiClient["default"].convertToType(data['digest_algo'], 'String');
        }

        if (data.hasOwnProperty('locality')) {
          obj['locality'] = _ApiClient["default"].convertToType(data['locality'], 'String');
        }

        if (data.hasOwnProperty('organization')) {
          obj['organization'] = _ApiClient["default"].convertToType(data['organization'], 'String');
        }

        if (data.hasOwnProperty('province')) {
          obj['province'] = _ApiClient["default"].convertToType(data['province'], 'String');
        }

        if (data.hasOwnProperty('self_signed_enabled')) {
          obj['self_signed_enabled'] = _ApiClient["default"].convertToType(data['self_signed_enabled'], 'Boolean');
        }

        if (data.hasOwnProperty('ttl')) {
          obj['ttl'] = _ApiClient["default"].convertToType(data['ttl'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return CertificateTemplateInfo;
}();
/**
 * @member {String} common_name
 */


CertificateTemplateInfo.prototype['common_name'] = undefined;
/**
 * @member {String} country
 */

CertificateTemplateInfo.prototype['country'] = undefined;
/**
 * @member {String} digest_algo
 */

CertificateTemplateInfo.prototype['digest_algo'] = undefined;
/**
 * @member {String} locality
 */

CertificateTemplateInfo.prototype['locality'] = undefined;
/**
 * @member {String} organization
 */

CertificateTemplateInfo.prototype['organization'] = undefined;
/**
 * @member {String} province
 */

CertificateTemplateInfo.prototype['province'] = undefined;
/**
 * @member {Boolean} self_signed_enabled
 */

CertificateTemplateInfo.prototype['self_signed_enabled'] = undefined;
/**
 * @member {Number} ttl
 */

CertificateTemplateInfo.prototype['ttl'] = undefined;
var _default = CertificateTemplateInfo;
exports["default"] = _default;
