// akeyless/dist/model/ChefTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ChefTargetDetails model module.
 * @module model/ChefTargetDetails
 * @version 3.3.16
 */
var ChefTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ChefTargetDetails</code>.
   * ChefTargetDetails
   * @alias module:model/ChefTargetDetails
   */
  function ChefTargetDetails() {
    _classCallCheck(this, ChefTargetDetails);

    ChefTargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ChefTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ChefTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ChefTargetDetails} obj Optional instance to populate.
     * @return {module:model/ChefTargetDetails} The populated <code>ChefTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ChefTargetDetails();

        if (data.hasOwnProperty('chef_server_host_name')) {
          obj['chef_server_host_name'] = _ApiClient["default"].convertToType(data['chef_server_host_name'], 'String');
        }

        if (data.hasOwnProperty('chef_server_key')) {
          obj['chef_server_key'] = _ApiClient["default"].convertToType(data['chef_server_key'], 'String');
        }

        if (data.hasOwnProperty('chef_server_port')) {
          obj['chef_server_port'] = _ApiClient["default"].convertToType(data['chef_server_port'], 'String');
        }

        if (data.hasOwnProperty('chef_server_url')) {
          obj['chef_server_url'] = _ApiClient["default"].convertToType(data['chef_server_url'], 'String');
        }

        if (data.hasOwnProperty('chef_server_username')) {
          obj['chef_server_username'] = _ApiClient["default"].convertToType(data['chef_server_username'], 'String');
        }

        if (data.hasOwnProperty('chef_skip_ssl')) {
          obj['chef_skip_ssl'] = _ApiClient["default"].convertToType(data['chef_skip_ssl'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return ChefTargetDetails;
}();
/**
 * @member {String} chef_server_host_name
 */


ChefTargetDetails.prototype['chef_server_host_name'] = undefined;
/**
 * @member {String} chef_server_key
 */

ChefTargetDetails.prototype['chef_server_key'] = undefined;
/**
 * @member {String} chef_server_port
 */

ChefTargetDetails.prototype['chef_server_port'] = undefined;
/**
 * @member {String} chef_server_url
 */

ChefTargetDetails.prototype['chef_server_url'] = undefined;
/**
 * @member {String} chef_server_username
 */

ChefTargetDetails.prototype['chef_server_username'] = undefined;
/**
 * @member {Boolean} chef_skip_ssl
 */

ChefTargetDetails.prototype['chef_skip_ssl'] = undefined;
var _default = ChefTargetDetails;
exports["default"] = _default;
