// akeyless/dist/model/CreateGroupOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateGroupOutput model module.
 * @module model/CreateGroupOutput
 * @version 3.5.0
 */
var CreateGroupOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateGroupOutput</code>.
   * @alias module:model/CreateGroupOutput
   */
  function CreateGroupOutput() {
    _classCallCheck(this, CreateGroupOutput);

    CreateGroupOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateGroupOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CreateGroupOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateGroupOutput} obj Optional instance to populate.
     * @return {module:model/CreateGroupOutput} The populated <code>CreateGroupOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateGroupOutput();

        if (data.hasOwnProperty('group_alias')) {
          obj['group_alias'] = _ApiClient["default"].convertToType(data['group_alias'], 'String');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateGroupOutput;
}();
/**
 * @member {String} group_alias
 */


CreateGroupOutput.prototype['group_alias'] = undefined;
/**
 * @member {String} id
 */

CreateGroupOutput.prototype['id'] = undefined;
/**
 * @member {String} name
 */

CreateGroupOutput.prototype['name'] = undefined;
var _default = CreateGroupOutput;
exports["default"] = _default;
