// akeyless/dist/model/CreateGroup.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateGroup model module.
 * @module model/CreateGroup
 * @version 3.5.0
 */
var CreateGroup = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateGroup</code>.
   * @alias module:model/CreateGroup
   * @param groupAlias {String} A short group alias
   * @param name {String} Group name
   */
  function CreateGroup(groupAlias, name) {
    _classCallCheck(this, CreateGroup);

    CreateGroup.initialize(this, groupAlias, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateGroup, null, [{
    key: "initialize",
    value: function initialize(obj, groupAlias, name) {
      obj['group-alias'] = groupAlias;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>CreateGroup</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateGroup} obj Optional instance to populate.
     * @return {module:model/CreateGroup} The populated <code>CreateGroup</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateGroup();

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('group-alias')) {
          obj['group-alias'] = _ApiClient["default"].convertToType(data['group-alias'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-assignment')) {
          obj['user-assignment'] = _ApiClient["default"].convertToType(data['user-assignment'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateGroup;
}();
/**
 * Description of the object
 * @member {String} description
 */


CreateGroup.prototype['description'] = undefined;
/**
 * A short group alias
 * @member {String} group-alias
 */

CreateGroup.prototype['group-alias'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

CreateGroup.prototype['json'] = false;
/**
 * Group name
 * @member {String} name
 */

CreateGroup.prototype['name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateGroup.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateGroup.prototype['uid-token'] = undefined;
/**
 * A json string defining the access permission assignment for this client
 * @member {String} user-assignment
 */

CreateGroup.prototype['user-assignment'] = undefined;
var _default = CreateGroup;
exports["default"] = _default;
