// akeyless/dist/model/CreateOidcApp.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateOidcApp model module.
 * @module model/CreateOidcApp
 * @version 3.5.0
 */
var CreateOidcApp = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateOidcApp</code>.
   * @alias module:model/CreateOidcApp
   * @param name {String} OIDC application name
   */
  function CreateOidcApp(name) {
    _classCallCheck(this, CreateOidcApp);

    CreateOidcApp.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateOidcApp, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>CreateOidcApp</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateOidcApp} obj Optional instance to populate.
     * @return {module:model/CreateOidcApp} The populated <code>CreateOidcApp</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateOidcApp();

        if (data.hasOwnProperty('accessibility')) {
          obj['accessibility'] = _ApiClient["default"].convertToType(data['accessibility'], 'String');
        }

        if (data.hasOwnProperty('audience')) {
          obj['audience'] = _ApiClient["default"].convertToType(data['audience'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('permission-assignment')) {
          obj['permission-assignment'] = _ApiClient["default"].convertToType(data['permission-assignment'], 'String');
        }

        if (data.hasOwnProperty('public')) {
          obj['public'] = _ApiClient["default"].convertToType(data['public'], 'Boolean');
        }

        if (data.hasOwnProperty('redirect-uris')) {
          obj['redirect-uris'] = _ApiClient["default"].convertToType(data['redirect-uris'], 'String');
        }

        if (data.hasOwnProperty('scopes')) {
          obj['scopes'] = _ApiClient["default"].convertToType(data['scopes'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateOidcApp;
}();
/**
 * for personal password manager
 * @member {String} accessibility
 * @default 'regular'
 */


CreateOidcApp.prototype['accessibility'] = 'regular';
/**
 * A comma separated list of allowed audiences
 * @member {String} audience
 */

CreateOidcApp.prototype['audience'] = undefined;
/**
 * Protection from accidental deletion of this item [true/false]
 * @member {String} delete_protection
 */

CreateOidcApp.prototype['delete_protection'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

CreateOidcApp.prototype['description'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

CreateOidcApp.prototype['json'] = false;
/**
 * The name of a key that used to encrypt the OIDC application (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

CreateOidcApp.prototype['key'] = undefined;
/**
 * Deprecated - use description
 * @member {String} metadata
 */

CreateOidcApp.prototype['metadata'] = undefined;
/**
 * OIDC application name
 * @member {String} name
 */

CreateOidcApp.prototype['name'] = undefined;
/**
 * A json string defining the access permission assignment for this app
 * @member {String} permission-assignment
 */

CreateOidcApp.prototype['permission-assignment'] = undefined;
/**
 * Set to true if the app is public (cannot keep secrets)
 * @member {Boolean} public
 */

CreateOidcApp.prototype['public'] = undefined;
/**
 * A comma separated list of allowed redirect uris
 * @member {String} redirect-uris
 */

CreateOidcApp.prototype['redirect-uris'] = undefined;
/**
 * A comma separated list of allowed scopes
 * @member {String} scopes
 * @default 'openid'
 */

CreateOidcApp.prototype['scopes'] = 'openid';
/**
 * Add tags attached to this object
 * @member {Array.<String>} tags
 */

CreateOidcApp.prototype['tags'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateOidcApp.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateOidcApp.prototype['uid-token'] = undefined;
var _default = CreateOidcApp;
exports["default"] = _default;
