// akeyless/dist/model/CreateESM.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateESM model module.
 * @module model/CreateESM
 * @version 3.3.14
 */
var CreateESM = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateESM</code>.
   * CreateESM is a command that creates an External Secrets Manager
   * @alias module:model/CreateESM
   * @param name {String} External Secrets Manager name
   * @param targetToAssociate {String} Target External Secrets Manager to connect
   */
  function CreateESM(name, targetToAssociate) {
    _classCallCheck(this, CreateESM);

    CreateESM.initialize(this, name, targetToAssociate);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateESM, null, [{
    key: "initialize",
    value: function initialize(obj, name, targetToAssociate) {
      obj['name'] = name;
      obj['target-to-associate'] = targetToAssociate;
    }
    /**
     * Constructs a <code>CreateESM</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateESM} obj Optional instance to populate.
     * @return {module:model/CreateESM} The populated <code>CreateESM</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateESM();

        if (data.hasOwnProperty('azure-kv-name')) {
          obj['azure-kv-name'] = _ApiClient["default"].convertToType(data['azure-kv-name'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('k8s-namespace')) {
          obj['k8s-namespace'] = _ApiClient["default"].convertToType(data['k8s-namespace'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-to-associate')) {
          obj['target-to-associate'] = _ApiClient["default"].convertToType(data['target-to-associate'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateESM;
}();
/**
 * Azure Key Vault name (Relevant only for Azure targets)
 * @member {String} azure-kv-name
 */


CreateESM.prototype['azure-kv-name'] = undefined;
/**
 * Protection from accidental deletion of this item [true/false]
 * @member {String} delete_protection
 */

CreateESM.prototype['delete_protection'] = undefined;
/**
 * Description of the External Secrets Manager
 * @member {String} description
 */

CreateESM.prototype['description'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

CreateESM.prototype['json'] = false;
/**
 * K8s namespace (Relevant to Kubernetes targets)
 * @member {String} k8s-namespace
 */

CreateESM.prototype['k8s-namespace'] = undefined;
/**
 * External Secrets Manager name
 * @member {String} name
 */

CreateESM.prototype['name'] = undefined;
/**
 * List of the tags attached to this External Secrets Manager
 * @member {Array.<String>} tags
 */

CreateESM.prototype['tags'] = undefined;
/**
 * Target External Secrets Manager to connect
 * @member {String} target-to-associate
 */

CreateESM.prototype['target-to-associate'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateESM.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateESM.prototype['uid-token'] = undefined;
var _default = CreateESM;
exports["default"] = _default;
