// akeyless/dist/model/CreateEventForwarder.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateEventForwarder model module.
 * @module model/CreateEventForwarder
 * @version 3.3.5
 */
var CreateEventForwarder = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateEventForwarder</code>.
   * @alias module:model/CreateEventForwarder
   * @param eventSourceLocations {Array.<String>} Event sources
   * @param forwarderType {String} 
   * @param name {String} EventForwarder name
   * @param runnerType {String} 
   */
  function CreateEventForwarder(eventSourceLocations, forwarderType, name, runnerType) {
    _classCallCheck(this, CreateEventForwarder);

    CreateEventForwarder.initialize(this, eventSourceLocations, forwarderType, name, runnerType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateEventForwarder, null, [{
    key: "initialize",
    value: function initialize(obj, eventSourceLocations, forwarderType, name, runnerType) {
      obj['event-source-locations'] = eventSourceLocations;
      obj['forwarder-type'] = forwarderType;
      obj['name'] = name;
      obj['runner-type'] = runnerType;
    }
    /**
     * Constructs a <code>CreateEventForwarder</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateEventForwarder} obj Optional instance to populate.
     * @return {module:model/CreateEventForwarder} The populated <code>CreateEventForwarder</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateEventForwarder();

        if (data.hasOwnProperty('admin-name')) {
          obj['admin-name'] = _ApiClient["default"].convertToType(data['admin-name'], 'String');
        }

        if (data.hasOwnProperty('admin-pwd')) {
          obj['admin-pwd'] = _ApiClient["default"].convertToType(data['admin-pwd'], 'String');
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('email-to')) {
          obj['email-to'] = _ApiClient["default"].convertToType(data['email-to'], 'String');
        }

        if (data.hasOwnProperty('event-source-locations')) {
          obj['event-source-locations'] = _ApiClient["default"].convertToType(data['event-source-locations'], ['String']);
        }

        if (data.hasOwnProperty('event-source-type')) {
          obj['event-source-type'] = _ApiClient["default"].convertToType(data['event-source-type'], 'String');
        }

        if (data.hasOwnProperty('event-types')) {
          obj['event-types'] = _ApiClient["default"].convertToType(data['event-types'], ['String']);
        }

        if (data.hasOwnProperty('every')) {
          obj['every'] = _ApiClient["default"].convertToType(data['every'], 'String');
        }

        if (data.hasOwnProperty('forwarder-type')) {
          obj['forwarder-type'] = _ApiClient["default"].convertToType(data['forwarder-type'], 'String');
        }

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('runner-type')) {
          obj['runner-type'] = _ApiClient["default"].convertToType(data['runner-type'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateEventForwarder;
}();
/**
 * Workstation Admin Name
 * @member {String} admin-name
 */


CreateEventForwarder.prototype['admin-name'] = undefined;
/**
 * Workstation Admin password
 * @member {String} admin-pwd
 */

CreateEventForwarder.prototype['admin-pwd'] = undefined;
/**
 * Deprecated - use description
 * @member {String} comment
 */

CreateEventForwarder.prototype['comment'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

CreateEventForwarder.prototype['description'] = undefined;
/**
 * A comma seperated list of email addresses to send event to (relevant only for \\\"email\\\" Event Forwarder)
 * @member {String} email-to
 */

CreateEventForwarder.prototype['email-to'] = undefined;
/**
 * Event sources
 * @member {Array.<String>} event-source-locations
 */

CreateEventForwarder.prototype['event-source-locations'] = undefined;
/**
 * Event Source type [item, target]
 * @member {String} event-source-type
 * @default 'item'
 */

CreateEventForwarder.prototype['event-source-type'] = 'item';
/**
 * Event types
 * @member {Array.<String>} event-types
 */

CreateEventForwarder.prototype['event-types'] = undefined;
/**
 * Rate of periodic runner repetition in hours
 * @member {String} every
 */

CreateEventForwarder.prototype['every'] = undefined;
/**
 * @member {String} forwarder-type
 */

CreateEventForwarder.prototype['forwarder-type'] = undefined;
/**
 * Workstation Host
 * @member {String} host
 */

CreateEventForwarder.prototype['host'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

CreateEventForwarder.prototype['json'] = false;
/**
 * The name of a key that used to encrypt the EventForwarder secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

CreateEventForwarder.prototype['key'] = undefined;
/**
 * EventForwarder name
 * @member {String} name
 */

CreateEventForwarder.prototype['name'] = undefined;
/**
 * @member {String} runner-type
 */

CreateEventForwarder.prototype['runner-type'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateEventForwarder.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateEventForwarder.prototype['uid-token'] = undefined;
var _default = CreateEventForwarder;
exports["default"] = _default;
