// akeyless/dist/model/EKSTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The EKSTargetDetails model module.
 * @module model/EKSTargetDetails
 * @version 3.3.16
 */
var EKSTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>EKSTargetDetails</code>.
   * EKSTargetDetails defines details related to connecting to a EKS (Elastic Container Service) target
   * @alias module:model/EKSTargetDetails
   */
  function EKSTargetDetails() {
    _classCallCheck(this, EKSTargetDetails);

    EKSTargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(EKSTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>EKSTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/EKSTargetDetails} obj Optional instance to populate.
     * @return {module:model/EKSTargetDetails} The populated <code>EKSTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new EKSTargetDetails();

        if (data.hasOwnProperty('eks_access_key_id')) {
          obj['eks_access_key_id'] = _ApiClient["default"].convertToType(data['eks_access_key_id'], 'String');
        }

        if (data.hasOwnProperty('eks_cluster_ca_certificate')) {
          obj['eks_cluster_ca_certificate'] = _ApiClient["default"].convertToType(data['eks_cluster_ca_certificate'], 'String');
        }

        if (data.hasOwnProperty('eks_cluster_endpoint')) {
          obj['eks_cluster_endpoint'] = _ApiClient["default"].convertToType(data['eks_cluster_endpoint'], 'String');
        }

        if (data.hasOwnProperty('eks_cluster_name')) {
          obj['eks_cluster_name'] = _ApiClient["default"].convertToType(data['eks_cluster_name'], 'String');
        }

        if (data.hasOwnProperty('eks_region')) {
          obj['eks_region'] = _ApiClient["default"].convertToType(data['eks_region'], 'String');
        }

        if (data.hasOwnProperty('eks_secret_access_key')) {
          obj['eks_secret_access_key'] = _ApiClient["default"].convertToType(data['eks_secret_access_key'], 'String');
        }

        if (data.hasOwnProperty('use_gw_cloud_identity')) {
          obj['use_gw_cloud_identity'] = _ApiClient["default"].convertToType(data['use_gw_cloud_identity'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return EKSTargetDetails;
}();
/**
 * @member {String} eks_access_key_id
 */


EKSTargetDetails.prototype['eks_access_key_id'] = undefined;
/**
 * @member {String} eks_cluster_ca_certificate
 */

EKSTargetDetails.prototype['eks_cluster_ca_certificate'] = undefined;
/**
 * @member {String} eks_cluster_endpoint
 */

EKSTargetDetails.prototype['eks_cluster_endpoint'] = undefined;
/**
 * @member {String} eks_cluster_name
 */

EKSTargetDetails.prototype['eks_cluster_name'] = undefined;
/**
 * @member {String} eks_region
 */

EKSTargetDetails.prototype['eks_region'] = undefined;
/**
 * @member {String} eks_secret_access_key
 */

EKSTargetDetails.prototype['eks_secret_access_key'] = undefined;
/**
 * @member {Boolean} use_gw_cloud_identity
 */

EKSTargetDetails.prototype['use_gw_cloud_identity'] = undefined;
var _default = EKSTargetDetails;
exports["default"] = _default;
