// akeyless/dist/model/EsmUpdate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The EsmUpdate model module.
 * @module model/EsmUpdate
 * @version 3.5.0
 */
var EsmUpdate = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>EsmUpdate</code>.
   * esmUpdate is a command that updates a secret in an External Secrets Manager
   * @alias module:model/EsmUpdate
   * @param esmName {String} Name of the External Secrets Manager item
   * @param secretId {String} The external secret id (or name, for AWS, Azure or K8s targets) to update
   * @param value {String} Value of the external secret item, either text or base64 encoded binary
   */
  function EsmUpdate(esmName, secretId, value) {
    _classCallCheck(this, EsmUpdate);

    EsmUpdate.initialize(this, esmName, secretId, value);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(EsmUpdate, null, [{
    key: "initialize",
    value: function initialize(obj, esmName, secretId, value) {
      obj['esm-name'] = esmName;
      obj['secret-id'] = secretId;
      obj['value'] = value;
    }
    /**
     * Constructs a <code>EsmUpdate</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/EsmUpdate} obj Optional instance to populate.
     * @return {module:model/EsmUpdate} The populated <code>EsmUpdate</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new EsmUpdate();

        if (data.hasOwnProperty('binary-value')) {
          obj['binary-value'] = _ApiClient["default"].convertToType(data['binary-value'], 'Boolean');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('esm-name')) {
          obj['esm-name'] = _ApiClient["default"].convertToType(data['esm-name'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('secret-id')) {
          obj['secret-id'] = _ApiClient["default"].convertToType(data['secret-id'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'String');
        }
      }

      return obj;
    }
  }]);

  return EsmUpdate;
}();
/**
 * Use this option if the external secret value is a base64 encoded binary
 * @member {Boolean} binary-value
 */


EsmUpdate.prototype['binary-value'] = undefined;
/**
 * Description of the external secret
 * @member {String} description
 */

EsmUpdate.prototype['description'] = undefined;
/**
 * Name of the External Secrets Manager item
 * @member {String} esm-name
 */

EsmUpdate.prototype['esm-name'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

EsmUpdate.prototype['json'] = false;
/**
 * The external secret id (or name, for AWS, Azure or K8s targets) to update
 * @member {String} secret-id
 */

EsmUpdate.prototype['secret-id'] = undefined;
/**
 * Tags for the external secret
 * @member {Object.<String, String>} tags
 */

EsmUpdate.prototype['tags'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

EsmUpdate.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

EsmUpdate.prototype['uid-token'] = undefined;
/**
 * Value of the external secret item, either text or base64 encoded binary
 * @member {String} value
 */

EsmUpdate.prototype['value'] = undefined;
var _default = EsmUpdate;
exports["default"] = _default;
