// akeyless/dist/model/EventAction.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The EventAction model module.
 * @module model/EventAction
 * @version 3.3.12
 */
var EventAction = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>EventAction</code>.
   * @alias module:model/EventAction
   * @param action {String} The Event Action [approve/deny]
   * @param eventId {Number} The Event ID
   */
  function EventAction(action, eventId) {
    _classCallCheck(this, EventAction);

    EventAction.initialize(this, action, eventId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(EventAction, null, [{
    key: "initialize",
    value: function initialize(obj, action, eventId) {
      obj['action'] = action;
      obj['event-id'] = eventId;
    }
    /**
     * Constructs a <code>EventAction</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/EventAction} obj Optional instance to populate.
     * @return {module:model/EventAction} The populated <code>EventAction</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new EventAction();

        if (data.hasOwnProperty('action')) {
          obj['action'] = _ApiClient["default"].convertToType(data['action'], 'String');
        }

        if (data.hasOwnProperty('event-id')) {
          obj['event-id'] = _ApiClient["default"].convertToType(data['event-id'], 'Number');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return EventAction;
}();
/**
 * The Event Action [approve/deny]
 * @member {String} action
 */


EventAction.prototype['action'] = undefined;
/**
 * The Event ID
 * @member {Number} event-id
 */

EventAction.prototype['event-id'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

EventAction.prototype['json'] = false;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

EventAction.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

EventAction.prototype['uid-token'] = undefined;
var _default = EventAction;
exports["default"] = _default;
