// akeyless/dist/model/GatewayCreateProducerNativeK8S.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayCreateProducerNativeK8S model module.
 * @module model/GatewayCreateProducerNativeK8S
 * @version 3.3.9
 */
var GatewayCreateProducerNativeK8S = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayCreateProducerNativeK8S</code>.
   * gatewayCreateProducerNativeK8S is a command that creates k8s producer
   * @alias module:model/GatewayCreateProducerNativeK8S
   * @param name {String} Producer name
   */
  function GatewayCreateProducerNativeK8S(name) {
    _classCallCheck(this, GatewayCreateProducerNativeK8S);

    GatewayCreateProducerNativeK8S.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayCreateProducerNativeK8S, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GatewayCreateProducerNativeK8S</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayCreateProducerNativeK8S} obj Optional instance to populate.
     * @return {module:model/GatewayCreateProducerNativeK8S} The populated <code>GatewayCreateProducerNativeK8S</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayCreateProducerNativeK8S();

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('k8s-allowed-namespaces')) {
          obj['k8s-allowed-namespaces'] = _ApiClient["default"].convertToType(data['k8s-allowed-namespaces'], 'String');
        }

        if (data.hasOwnProperty('k8s-cluster-ca-cert')) {
          obj['k8s-cluster-ca-cert'] = _ApiClient["default"].convertToType(data['k8s-cluster-ca-cert'], 'String');
        }

        if (data.hasOwnProperty('k8s-cluster-endpoint')) {
          obj['k8s-cluster-endpoint'] = _ApiClient["default"].convertToType(data['k8s-cluster-endpoint'], 'String');
        }

        if (data.hasOwnProperty('k8s-cluster-token')) {
          obj['k8s-cluster-token'] = _ApiClient["default"].convertToType(data['k8s-cluster-token'], 'String');
        }

        if (data.hasOwnProperty('k8s-namespace')) {
          obj['k8s-namespace'] = _ApiClient["default"].convertToType(data['k8s-namespace'], 'String');
        }

        if (data.hasOwnProperty('k8s-predefined-role-name')) {
          obj['k8s-predefined-role-name'] = _ApiClient["default"].convertToType(data['k8s-predefined-role-name'], 'String');
        }

        if (data.hasOwnProperty('k8s-predefined-role-type')) {
          obj['k8s-predefined-role-type'] = _ApiClient["default"].convertToType(data['k8s-predefined-role-type'], 'String');
        }

        if (data.hasOwnProperty('k8s-rolebinding-yaml-def')) {
          obj['k8s-rolebinding-yaml-def'] = _ApiClient["default"].convertToType(data['k8s-rolebinding-yaml-def'], 'String');
        }

        if (data.hasOwnProperty('k8s-service-account')) {
          obj['k8s-service-account'] = _ApiClient["default"].convertToType(data['k8s-service-account'], 'String');
        }

        if (data.hasOwnProperty('k8s-service-account-type')) {
          obj['k8s-service-account-type'] = _ApiClient["default"].convertToType(data['k8s-service-account-type'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('secure-access-allow-port-forwading')) {
          obj['secure-access-allow-port-forwading'] = _ApiClient["default"].convertToType(data['secure-access-allow-port-forwading'], 'Boolean');
        }

        if (data.hasOwnProperty('secure-access-bastion-issuer')) {
          obj['secure-access-bastion-issuer'] = _ApiClient["default"].convertToType(data['secure-access-bastion-issuer'], 'String');
        }

        if (data.hasOwnProperty('secure-access-cluster-endpoint')) {
          obj['secure-access-cluster-endpoint'] = _ApiClient["default"].convertToType(data['secure-access-cluster-endpoint'], 'String');
        }

        if (data.hasOwnProperty('secure-access-dashboard-url')) {
          obj['secure-access-dashboard-url'] = _ApiClient["default"].convertToType(data['secure-access-dashboard-url'], 'String');
        }

        if (data.hasOwnProperty('secure-access-enable')) {
          obj['secure-access-enable'] = _ApiClient["default"].convertToType(data['secure-access-enable'], 'String');
        }

        if (data.hasOwnProperty('secure-access-web')) {
          obj['secure-access-web'] = _ApiClient["default"].convertToType(data['secure-access-web'], 'Boolean');
        }

        if (data.hasOwnProperty('secure-access-web-browsing')) {
          obj['secure-access-web-browsing'] = _ApiClient["default"].convertToType(data['secure-access-web-browsing'], 'Boolean');
        }

        if (data.hasOwnProperty('secure-access-web-proxy')) {
          obj['secure-access-web-proxy'] = _ApiClient["default"].convertToType(data['secure-access-web-proxy'], 'Boolean');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('use-gw-service-account')) {
          obj['use-gw-service-account'] = _ApiClient["default"].convertToType(data['use-gw-service-account'], 'Boolean');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayCreateProducerNativeK8S;
}();
/**
 * Protection from accidental deletion of this item [true/false]
 * @member {String} delete_protection
 */


GatewayCreateProducerNativeK8S.prototype['delete_protection'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GatewayCreateProducerNativeK8S.prototype['json'] = false;
/**
 * Comma-separated list of allowed K8S namespaces for the generated ServiceAccount (relevant only for k8s-service-account-type=dynamic)
 * @member {String} k8s-allowed-namespaces
 */

GatewayCreateProducerNativeK8S.prototype['k8s-allowed-namespaces'] = undefined;
/**
 * K8S cluster CA certificate
 * @member {String} k8s-cluster-ca-cert
 */

GatewayCreateProducerNativeK8S.prototype['k8s-cluster-ca-cert'] = undefined;
/**
 * K8S cluster URL endpoint
 * @member {String} k8s-cluster-endpoint
 */

GatewayCreateProducerNativeK8S.prototype['k8s-cluster-endpoint'] = undefined;
/**
 * K8S cluster Bearer token
 * @member {String} k8s-cluster-token
 */

GatewayCreateProducerNativeK8S.prototype['k8s-cluster-token'] = undefined;
/**
 * K8S Namespace where the ServiceAccount exists.
 * @member {String} k8s-namespace
 */

GatewayCreateProducerNativeK8S.prototype['k8s-namespace'] = undefined;
/**
 * The pre-existing Role or ClusterRole name to bind the generated ServiceAccount to (relevant only for k8s-service-account-type=dynamic)
 * @member {String} k8s-predefined-role-name
 */

GatewayCreateProducerNativeK8S.prototype['k8s-predefined-role-name'] = undefined;
/**
 * Specifies the type of the pre-existing K8S role [Role, ClusterRole] (relevant only for k8s-service-account-type=dynamic)
 * @member {String} k8s-predefined-role-type
 */

GatewayCreateProducerNativeK8S.prototype['k8s-predefined-role-type'] = undefined;
/**
 * Path to yaml file that contains definitions of K8S role and role binding (relevant only for k8s-service-account-type=dynamic)
 * @member {String} k8s-rolebinding-yaml-def
 */

GatewayCreateProducerNativeK8S.prototype['k8s-rolebinding-yaml-def'] = undefined;
/**
 * K8S ServiceAccount to extract token from.
 * @member {String} k8s-service-account
 */

GatewayCreateProducerNativeK8S.prototype['k8s-service-account'] = undefined;
/**
 * K8S ServiceAccount type [fixed, dynamic].
 * @member {String} k8s-service-account-type
 */

GatewayCreateProducerNativeK8S.prototype['k8s-service-account-type'] = undefined;
/**
 * Producer name
 * @member {String} name
 */

GatewayCreateProducerNativeK8S.prototype['name'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayCreateProducerNativeK8S.prototype['producer-encryption-key-name'] = undefined;
/**
 * Enable Port forwarding while using CLI access
 * @member {Boolean} secure-access-allow-port-forwading
 */

GatewayCreateProducerNativeK8S.prototype['secure-access-allow-port-forwading'] = undefined;
/**
 * Path to the SSH Certificate Issuer for your Akeyless Bastion
 * @member {String} secure-access-bastion-issuer
 */

GatewayCreateProducerNativeK8S.prototype['secure-access-bastion-issuer'] = undefined;
/**
 * The K8s cluster endpoint URL
 * @member {String} secure-access-cluster-endpoint
 */

GatewayCreateProducerNativeK8S.prototype['secure-access-cluster-endpoint'] = undefined;
/**
 * The K8s dashboard url
 * @member {String} secure-access-dashboard-url
 */

GatewayCreateProducerNativeK8S.prototype['secure-access-dashboard-url'] = undefined;
/**
 * Enable/Disable secure remote access [true/false]
 * @member {String} secure-access-enable
 */

GatewayCreateProducerNativeK8S.prototype['secure-access-enable'] = undefined;
/**
 * Enable Web Secure Remote Access
 * @member {Boolean} secure-access-web
 * @default false
 */

GatewayCreateProducerNativeK8S.prototype['secure-access-web'] = false;
/**
 * Secure browser via Akeyless Web Access Bastion
 * @member {Boolean} secure-access-web-browsing
 * @default false
 */

GatewayCreateProducerNativeK8S.prototype['secure-access-web-browsing'] = false;
/**
 * Web-Proxy via Akeyless Web Access Bastion
 * @member {Boolean} secure-access-web-proxy
 * @default false
 */

GatewayCreateProducerNativeK8S.prototype['secure-access-web-proxy'] = false;
/**
 * Add tags attached to this object
 * @member {Array.<String>} tags
 */

GatewayCreateProducerNativeK8S.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayCreateProducerNativeK8S.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayCreateProducerNativeK8S.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayCreateProducerNativeK8S.prototype['uid-token'] = undefined;
/**
 * Use the GW's service account
 * @member {Boolean} use-gw-service-account
 */

GatewayCreateProducerNativeK8S.prototype['use-gw-service-account'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayCreateProducerNativeK8S.prototype['user-ttl'] = '60m';
var _default = GatewayCreateProducerNativeK8S;
exports["default"] = _default;
