// akeyless/dist/model/GatewayBasicInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayBasicInfo model module.
 * @module model/GatewayBasicInfo
 * @version 3.3.5
 */
var GatewayBasicInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayBasicInfo</code>.
   * @alias module:model/GatewayBasicInfo
   */
  function GatewayBasicInfo() {
    _classCallCheck(this, GatewayBasicInfo);

    GatewayBasicInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayBasicInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GatewayBasicInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayBasicInfo} obj Optional instance to populate.
     * @return {module:model/GatewayBasicInfo} The populated <code>GatewayBasicInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayBasicInfo();

        if (data.hasOwnProperty('cluster_display_name')) {
          obj['cluster_display_name'] = _ApiClient["default"].convertToType(data['cluster_display_name'], 'String');
        }

        if (data.hasOwnProperty('cluster_id')) {
          obj['cluster_id'] = _ApiClient["default"].convertToType(data['cluster_id'], 'Number');
        }

        if (data.hasOwnProperty('cluster_name')) {
          obj['cluster_name'] = _ApiClient["default"].convertToType(data['cluster_name'], 'String');
        }

        if (data.hasOwnProperty('cluster_url')) {
          obj['cluster_url'] = _ApiClient["default"].convertToType(data['cluster_url'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayBasicInfo;
}();
/**
 * @member {String} cluster_display_name
 */


GatewayBasicInfo.prototype['cluster_display_name'] = undefined;
/**
 * @member {Number} cluster_id
 */

GatewayBasicInfo.prototype['cluster_id'] = undefined;
/**
 * @member {String} cluster_name
 */

GatewayBasicInfo.prototype['cluster_name'] = undefined;
/**
 * @member {String} cluster_url
 */

GatewayBasicInfo.prototype['cluster_url'] = undefined;
var _default = GatewayBasicInfo;
exports["default"] = _default;
