// akeyless/dist/model/GcpTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GcpTargetDetails model module.
 * @module model/GcpTargetDetails
 * @version 3.5.0
 */
var GcpTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GcpTargetDetails</code>.
   * @alias module:model/GcpTargetDetails
   */
  function GcpTargetDetails() {
    _classCallCheck(this, GcpTargetDetails);

    GcpTargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GcpTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GcpTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GcpTargetDetails} obj Optional instance to populate.
     * @return {module:model/GcpTargetDetails} The populated <code>GcpTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GcpTargetDetails();

        if (data.hasOwnProperty('gcp_service_account_email')) {
          obj['gcp_service_account_email'] = _ApiClient["default"].convertToType(data['gcp_service_account_email'], 'String');
        }

        if (data.hasOwnProperty('gcp_service_account_key')) {
          obj['gcp_service_account_key'] = _ApiClient["default"].convertToType(data['gcp_service_account_key'], 'String');
        }

        if (data.hasOwnProperty('gcp_service_account_key_base64')) {
          obj['gcp_service_account_key_base64'] = _ApiClient["default"].convertToType(data['gcp_service_account_key_base64'], 'String');
        }

        if (data.hasOwnProperty('use_gw_cloud_identity')) {
          obj['use_gw_cloud_identity'] = _ApiClient["default"].convertToType(data['use_gw_cloud_identity'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return GcpTargetDetails;
}();
/**
 * deprecated
 * @member {String} gcp_service_account_email
 */


GcpTargetDetails.prototype['gcp_service_account_email'] = undefined;
/**
 * @member {String} gcp_service_account_key
 */

GcpTargetDetails.prototype['gcp_service_account_key'] = undefined;
/**
 * @member {String} gcp_service_account_key_base64
 */

GcpTargetDetails.prototype['gcp_service_account_key_base64'] = undefined;
/**
 * @member {Boolean} use_gw_cloud_identity
 */

GcpTargetDetails.prototype['use_gw_cloud_identity'] = undefined;
var _default = GcpTargetDetails;
exports["default"] = _default;
