// akeyless/dist/model/GetCertificateValue.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetCertificateValue model module.
 * @module model/GetCertificateValue
 * @version 3.3.5
 */
var GetCertificateValue = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetCertificateValue</code>.
   * @alias module:model/GetCertificateValue
   * @param name {String} Certificate name
   */
  function GetCertificateValue(name) {
    _classCallCheck(this, GetCertificateValue);

    GetCertificateValue.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetCertificateValue, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GetCertificateValue</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetCertificateValue} obj Optional instance to populate.
     * @return {module:model/GetCertificateValue} The populated <code>GetCertificateValue</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetCertificateValue();

        if (data.hasOwnProperty('cert-issuer-name')) {
          obj['cert-issuer-name'] = _ApiClient["default"].convertToType(data['cert-issuer-name'], 'String');
        }

        if (data.hasOwnProperty('certificate-file-output')) {
          obj['certificate-file-output'] = _ApiClient["default"].convertToType(data['certificate-file-output'], 'String');
        }

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('issuance-token')) {
          obj['issuance-token'] = _ApiClient["default"].convertToType(data['issuance-token'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('private-key-file-output')) {
          obj['private-key-file-output'] = _ApiClient["default"].convertToType(data['private-key-file-output'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return GetCertificateValue;
}();
/**
 * The parent PKI Certificate Issuer's name of the certificate, required when used with display-id and token
 * @member {String} cert-issuer-name
 */


GetCertificateValue.prototype['cert-issuer-name'] = undefined;
/**
 * File to write the certificates to.
 * @member {String} certificate-file-output
 */

GetCertificateValue.prototype['certificate-file-output'] = undefined;
/**
 * Certificate display ID
 * @member {String} display-id
 */

GetCertificateValue.prototype['display-id'] = undefined;
/**
 * Token for getting the issued certificate
 * @member {String} issuance-token
 */

GetCertificateValue.prototype['issuance-token'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GetCertificateValue.prototype['json'] = false;
/**
 * Certificate name
 * @member {String} name
 * @default 'dummy_certificate_name'
 */

GetCertificateValue.prototype['name'] = 'dummy_certificate_name';
/**
 * File to write the private key to.
 * @member {String} private-key-file-output
 */

GetCertificateValue.prototype['private-key-file-output'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GetCertificateValue.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GetCertificateValue.prototype['uid-token'] = undefined;
/**
 * Certificate version
 * @member {Number} version
 */

GetCertificateValue.prototype['version'] = undefined;
var _default = GetCertificateValue;
exports["default"] = _default;
