// akeyless/dist/model/GetGroupOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AccessPermissionAssignment = _interopRequireDefault(require("./AccessPermissionAssignment"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetGroupOutput model module.
 * @module model/GetGroupOutput
 * @version 3.5.0
 */
var GetGroupOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetGroupOutput</code>.
   * @alias module:model/GetGroupOutput
   */
  function GetGroupOutput() {
    _classCallCheck(this, GetGroupOutput);

    GetGroupOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetGroupOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetGroupOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetGroupOutput} obj Optional instance to populate.
     * @return {module:model/GetGroupOutput} The populated <code>GetGroupOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetGroupOutput();

        if (data.hasOwnProperty('account_id')) {
          obj['account_id'] = _ApiClient["default"].convertToType(data['account_id'], 'String');
        }

        if (data.hasOwnProperty('creation_date')) {
          obj['creation_date'] = _ApiClient["default"].convertToType(data['creation_date'], 'Date');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('group_alias')) {
          obj['group_alias'] = _ApiClient["default"].convertToType(data['group_alias'], 'String');
        }

        if (data.hasOwnProperty('group_id')) {
          obj['group_id'] = _ApiClient["default"].convertToType(data['group_id'], 'String');
        }

        if (data.hasOwnProperty('group_name')) {
          obj['group_name'] = _ApiClient["default"].convertToType(data['group_name'], 'String');
        }

        if (data.hasOwnProperty('modification_date')) {
          obj['modification_date'] = _ApiClient["default"].convertToType(data['modification_date'], 'Date');
        }

        if (data.hasOwnProperty('user_assignments')) {
          obj['user_assignments'] = _ApiClient["default"].convertToType(data['user_assignments'], [_AccessPermissionAssignment["default"]]);
        }
      }

      return obj;
    }
  }]);

  return GetGroupOutput;
}();
/**
 * @member {String} account_id
 */


GetGroupOutput.prototype['account_id'] = undefined;
/**
 * @member {Date} creation_date
 */

GetGroupOutput.prototype['creation_date'] = undefined;
/**
 * @member {String} description
 */

GetGroupOutput.prototype['description'] = undefined;
/**
 * @member {String} group_alias
 */

GetGroupOutput.prototype['group_alias'] = undefined;
/**
 * @member {String} group_id
 */

GetGroupOutput.prototype['group_id'] = undefined;
/**
 * @member {String} group_name
 */

GetGroupOutput.prototype['group_name'] = undefined;
/**
 * @member {Date} modification_date
 */

GetGroupOutput.prototype['modification_date'] = undefined;
/**
 * @member {Array.<module:model/AccessPermissionAssignment>} user_assignments
 */

GetGroupOutput.prototype['user_assignments'] = undefined;
var _default = GetGroupOutput;
exports["default"] = _default;
