// akeyless/dist/model/GetUserEventStatusOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetUserEventStatusOutput model module.
 * @module model/GetUserEventStatusOutput
 * @version 3.5.0
 */
var GetUserEventStatusOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetUserEventStatusOutput</code>.
   * @alias module:model/GetUserEventStatusOutput
   */
  function GetUserEventStatusOutput() {
    _classCallCheck(this, GetUserEventStatusOutput);

    GetUserEventStatusOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetUserEventStatusOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetUserEventStatusOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetUserEventStatusOutput} obj Optional instance to populate.
     * @return {module:model/GetUserEventStatusOutput} The populated <code>GetUserEventStatusOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetUserEventStatusOutput();

        if (data.hasOwnProperty('access_status')) {
          obj['access_status'] = _ApiClient["default"].convertToType(data['access_status'], 'String');
        }

        if (data.hasOwnProperty('event_created_at')) {
          obj['event_created_at'] = _ApiClient["default"].convertToType(data['event_created_at'], 'Date');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GetUserEventStatusOutput;
}();
/**
 * @member {String} access_status
 */


GetUserEventStatusOutput.prototype['access_status'] = undefined;
/**
 * @member {Date} event_created_at
 */

GetUserEventStatusOutput.prototype['event_created_at'] = undefined;
/**
 * @member {String} status
 */

GetUserEventStatusOutput.prototype['status'] = undefined;
var _default = GetUserEventStatusOutput;
exports["default"] = _default;
