// akeyless/dist/model/GithubTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GithubTargetDetails model module.
 * @module model/GithubTargetDetails
 * @version 3.5.0
 */
var GithubTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GithubTargetDetails</code>.
   * @alias module:model/GithubTargetDetails
   */
  function GithubTargetDetails() {
    _classCallCheck(this, GithubTargetDetails);

    GithubTargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GithubTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GithubTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GithubTargetDetails} obj Optional instance to populate.
     * @return {module:model/GithubTargetDetails} The populated <code>GithubTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GithubTargetDetails();

        if (data.hasOwnProperty('github_app_id')) {
          obj['github_app_id'] = _ApiClient["default"].convertToType(data['github_app_id'], 'Number');
        }

        if (data.hasOwnProperty('github_app_private_key')) {
          obj['github_app_private_key'] = _ApiClient["default"].convertToType(data['github_app_private_key'], 'String');
        }

        if (data.hasOwnProperty('github_base_url')) {
          obj['github_base_url'] = _ApiClient["default"].convertToType(data['github_base_url'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GithubTargetDetails;
}();
/**
 * @member {Number} github_app_id
 */


GithubTargetDetails.prototype['github_app_id'] = undefined;
/**
 * @member {String} github_app_private_key
 */

GithubTargetDetails.prototype['github_app_private_key'] = undefined;
/**
 * @member {String} github_base_url
 */

GithubTargetDetails.prototype['github_base_url'] = undefined;
var _default = GithubTargetDetails;
exports["default"] = _default;
