// akeyless/dist/model/GlobalSignAtlasTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GlobalSignAtlasTargetDetails model module.
 * @module model/GlobalSignAtlasTargetDetails
 * @version 3.3.16
 */
var GlobalSignAtlasTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GlobalSignAtlasTargetDetails</code>.
   * GlobalSignAtlasTargetDetails
   * @alias module:model/GlobalSignAtlasTargetDetails
   */
  function GlobalSignAtlasTargetDetails() {
    _classCallCheck(this, GlobalSignAtlasTargetDetails);

    GlobalSignAtlasTargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GlobalSignAtlasTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GlobalSignAtlasTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GlobalSignAtlasTargetDetails} obj Optional instance to populate.
     * @return {module:model/GlobalSignAtlasTargetDetails} The populated <code>GlobalSignAtlasTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GlobalSignAtlasTargetDetails();

        if (data.hasOwnProperty('api_key')) {
          obj['api_key'] = _ApiClient["default"].convertToType(data['api_key'], 'String');
        }

        if (data.hasOwnProperty('api_secret')) {
          obj['api_secret'] = _ApiClient["default"].convertToType(data['api_secret'], 'String');
        }

        if (data.hasOwnProperty('mtls_cert')) {
          obj['mtls_cert'] = _ApiClient["default"].convertToType(data['mtls_cert'], 'String');
        }

        if (data.hasOwnProperty('mtls_key')) {
          obj['mtls_key'] = _ApiClient["default"].convertToType(data['mtls_key'], 'String');
        }

        if (data.hasOwnProperty('timeout')) {
          obj['timeout'] = _ApiClient["default"].convertToType(data['timeout'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return GlobalSignAtlasTargetDetails;
}();
/**
 * @member {String} api_key
 */


GlobalSignAtlasTargetDetails.prototype['api_key'] = undefined;
/**
 * @member {String} api_secret
 */

GlobalSignAtlasTargetDetails.prototype['api_secret'] = undefined;
/**
 * @member {String} mtls_cert
 */

GlobalSignAtlasTargetDetails.prototype['mtls_cert'] = undefined;
/**
 * @member {String} mtls_key
 */

GlobalSignAtlasTargetDetails.prototype['mtls_key'] = undefined;
/**
 * A Duration represents the elapsed time between two instants as an int64 nanosecond count. The representation limits the largest representable duration to approximately 290 years.
 * @member {Number} timeout
 */

GlobalSignAtlasTargetDetails.prototype['timeout'] = undefined;
var _default = GlobalSignAtlasTargetDetails;
exports["default"] = _default;
