// akeyless/dist/model/GoogleChronicleForwardingConfig.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GoogleChronicleForwardingConfig model module.
 * @module model/GoogleChronicleForwardingConfig
 * @version 3.5.0
 */
var GoogleChronicleForwardingConfig = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GoogleChronicleForwardingConfig</code>.
   * @alias module:model/GoogleChronicleForwardingConfig
   */
  function GoogleChronicleForwardingConfig() {
    _classCallCheck(this, GoogleChronicleForwardingConfig);

    GoogleChronicleForwardingConfig.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GoogleChronicleForwardingConfig, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GoogleChronicleForwardingConfig</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GoogleChronicleForwardingConfig} obj Optional instance to populate.
     * @return {module:model/GoogleChronicleForwardingConfig} The populated <code>GoogleChronicleForwardingConfig</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GoogleChronicleForwardingConfig();

        if (data.hasOwnProperty('customer_id')) {
          obj['customer_id'] = _ApiClient["default"].convertToType(data['customer_id'], 'String');
        }

        if (data.hasOwnProperty('log_type')) {
          obj['log_type'] = _ApiClient["default"].convertToType(data['log_type'], 'String');
        }

        if (data.hasOwnProperty('region')) {
          obj['region'] = _ApiClient["default"].convertToType(data['region'], 'String');
        }

        if (data.hasOwnProperty('service_account_key')) {
          obj['service_account_key'] = _ApiClient["default"].convertToType(data['service_account_key'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GoogleChronicleForwardingConfig;
}();
/**
 * @member {String} customer_id
 */


GoogleChronicleForwardingConfig.prototype['customer_id'] = undefined;
/**
 * @member {String} log_type
 */

GoogleChronicleForwardingConfig.prototype['log_type'] = undefined;
/**
 * @member {String} region
 */

GoogleChronicleForwardingConfig.prototype['region'] = undefined;
/**
 * @member {String} service_account_key
 */

GoogleChronicleForwardingConfig.prototype['service_account_key'] = undefined;
var _default = GoogleChronicleForwardingConfig;
exports["default"] = _default;
