// akeyless/dist/model/Hmac.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Hmac model module.
 * @module model/Hmac
 * @version 3.3.9
 */
var Hmac = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Hmac</code>.
   * @alias module:model/Hmac
   * @param keyName {String} The name of the key to use in the encryption process
   */
  function Hmac(keyName) {
    _classCallCheck(this, Hmac);

    Hmac.initialize(this, keyName);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Hmac, null, [{
    key: "initialize",
    value: function initialize(obj, keyName) {
      obj['key-name'] = keyName;
    }
    /**
     * Constructs a <code>Hmac</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Hmac} obj Optional instance to populate.
     * @return {module:model/Hmac} The populated <code>Hmac</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Hmac();

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('hash-function')) {
          obj['hash-function'] = _ApiClient["default"].convertToType(data['hash-function'], 'String');
        }

        if (data.hasOwnProperty('input-format')) {
          obj['input-format'] = _ApiClient["default"].convertToType(data['input-format'], 'String');
        }

        if (data.hasOwnProperty('item-id')) {
          obj['item-id'] = _ApiClient["default"].convertToType(data['item-id'], 'Number');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key-name')) {
          obj['key-name'] = _ApiClient["default"].convertToType(data['key-name'], 'String');
        }

        if (data.hasOwnProperty('plaintext')) {
          obj['plaintext'] = _ApiClient["default"].convertToType(data['plaintext'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return Hmac;
}();
/**
 * The display id of the key to use in the encryption process
 * @member {String} display-id
 */


Hmac.prototype['display-id'] = undefined;
/**
 * Hash function [sha-256,sha-512]
 * @member {String} hash-function
 * @default 'sha-256'
 */

Hmac.prototype['hash-function'] = 'sha-256';
/**
 * Select default assumed format for any plaintext input. Currently supported options: [base64]
 * @member {String} input-format
 */

Hmac.prototype['input-format'] = undefined;
/**
 * The item id of the key to use in the encryption process
 * @member {Number} item-id
 */

Hmac.prototype['item-id'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

Hmac.prototype['json'] = false;
/**
 * The name of the key to use in the encryption process
 * @member {String} key-name
 */

Hmac.prototype['key-name'] = undefined;
/**
 * Data to perform hmac on
 * @member {String} plaintext
 */

Hmac.prototype['plaintext'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

Hmac.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

Hmac.prototype['uid-token'] = undefined;
var _default = Hmac;
exports["default"] = _default;
