// akeyless/dist/model/ImportPasswords.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ImportPasswords model module.
 * @module model/ImportPasswords
 * @version 3.5.0
 */
var ImportPasswords = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ImportPasswords</code>.
   * importPasswords is a command that import passwords
   * @alias module:model/ImportPasswords
   * @param importPath {String} File path
   */
  function ImportPasswords(importPath) {
    _classCallCheck(this, ImportPasswords);

    ImportPasswords.initialize(this, importPath);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ImportPasswords, null, [{
    key: "initialize",
    value: function initialize(obj, importPath) {
      obj['import-path'] = importPath;
    }
    /**
     * Constructs a <code>ImportPasswords</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ImportPasswords} obj Optional instance to populate.
     * @return {module:model/ImportPasswords} The populated <code>ImportPasswords</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ImportPasswords();

        if (data.hasOwnProperty('accessibility')) {
          obj['accessibility'] = _ApiClient["default"].convertToType(data['accessibility'], 'String');
        }

        if (data.hasOwnProperty('format')) {
          obj['format'] = _ApiClient["default"].convertToType(data['format'], 'String');
        }

        if (data.hasOwnProperty('import-path')) {
          obj['import-path'] = _ApiClient["default"].convertToType(data['import-path'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('protection_key')) {
          obj['protection_key'] = _ApiClient["default"].convertToType(data['protection_key'], 'String');
        }

        if (data.hasOwnProperty('target-folder')) {
          obj['target-folder'] = _ApiClient["default"].convertToType(data['target-folder'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ImportPasswords;
}();
/**
 * for personal password manager
 * @member {String} accessibility
 * @default 'personal'
 */


ImportPasswords.prototype['accessibility'] = 'personal';
/**
 * Password format type [LastPass/Chrome/Firefox]
 * @member {String} format
 * @default 'LastPass'
 */

ImportPasswords.prototype['format'] = 'LastPass';
/**
 * File path
 * @member {String} import-path
 */

ImportPasswords.prototype['import-path'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

ImportPasswords.prototype['json'] = false;
/**
 * The name of a key that used to encrypt the secret value (if empty, the account default protectionKey key will be used)
 * @member {String} protection_key
 */

ImportPasswords.prototype['protection_key'] = undefined;
/**
 * Target folder for imported passwords
 * @member {String} target-folder
 * @default '/'
 */

ImportPasswords.prototype['target-folder'] = '/';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

ImportPasswords.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

ImportPasswords.prototype['uid-token'] = undefined;
var _default = ImportPasswords;
exports["default"] = _default;
