// akeyless/dist/model/ImportPasswordsOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ImportPasswordsOutput model module.
 * @module model/ImportPasswordsOutput
 * @version 3.5.0
 */
var ImportPasswordsOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ImportPasswordsOutput</code>.
   * @alias module:model/ImportPasswordsOutput
   */
  function ImportPasswordsOutput() {
    _classCallCheck(this, ImportPasswordsOutput);

    ImportPasswordsOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ImportPasswordsOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ImportPasswordsOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ImportPasswordsOutput} obj Optional instance to populate.
     * @return {module:model/ImportPasswordsOutput} The populated <code>ImportPasswordsOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ImportPasswordsOutput();

        if (data.hasOwnProperty('imported')) {
          obj['imported'] = _ApiClient["default"].convertToType(data['imported'], 'Number');
        }

        if (data.hasOwnProperty('passwords_in_file')) {
          obj['passwords_in_file'] = _ApiClient["default"].convertToType(data['passwords_in_file'], 'Number');
        }

        if (data.hasOwnProperty('successfully_parsed')) {
          obj['successfully_parsed'] = _ApiClient["default"].convertToType(data['successfully_parsed'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return ImportPasswordsOutput;
}();
/**
 * @member {Number} imported
 */


ImportPasswordsOutput.prototype['imported'] = undefined;
/**
 * @member {Number} passwords_in_file
 */

ImportPasswordsOutput.prototype['passwords_in_file'] = undefined;
/**
 * @member {Number} successfully_parsed
 */

ImportPasswordsOutput.prototype['successfully_parsed'] = undefined;
var _default = ImportPasswordsOutput;
exports["default"] = _default;
