// akeyless/dist/model/ItemGeneralInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CertificateChainInfo = _interopRequireDefault(require("./CertificateChainInfo"));

var _CertificateIssueInfo = _interopRequireDefault(require("./CertificateIssueInfo"));

var _CertificateTemplateInfo = _interopRequireDefault(require("./CertificateTemplateInfo"));

var _ClassicKeyDetailsInfo = _interopRequireDefault(require("./ClassicKeyDetailsInfo"));

var _DynamicSecretProducerInfo = _interopRequireDefault(require("./DynamicSecretProducerInfo"));

var _ImporterInfo = _interopRequireDefault(require("./ImporterInfo"));

var _OidcClientInfo = _interopRequireDefault(require("./OidcClientInfo"));

var _PasswordPolicyInfo = _interopRequireDefault(require("./PasswordPolicyInfo"));

var _RotatedSecretDetailsInfo = _interopRequireDefault(require("./RotatedSecretDetailsInfo"));

var _SecureRemoteAccess = _interopRequireDefault(require("./SecureRemoteAccess"));

var _StaticSecretDetailsInfo = _interopRequireDefault(require("./StaticSecretDetailsInfo"));

var _TokenizerInfo = _interopRequireDefault(require("./TokenizerInfo"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ItemGeneralInfo model module.
 * @module model/ItemGeneralInfo
 * @version 3.5.0
 */
var ItemGeneralInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ItemGeneralInfo</code>.
   * @alias module:model/ItemGeneralInfo
   */
  function ItemGeneralInfo() {
    _classCallCheck(this, ItemGeneralInfo);

    ItemGeneralInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ItemGeneralInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ItemGeneralInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ItemGeneralInfo} obj Optional instance to populate.
     * @return {module:model/ItemGeneralInfo} The populated <code>ItemGeneralInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ItemGeneralInfo();

        if (data.hasOwnProperty('cert_issue_details')) {
          obj['cert_issue_details'] = _CertificateIssueInfo["default"].constructFromObject(data['cert_issue_details']);
        }

        if (data.hasOwnProperty('certificate_chain_info')) {
          obj['certificate_chain_info'] = _CertificateChainInfo["default"].constructFromObject(data['certificate_chain_info']);
        }

        if (data.hasOwnProperty('certificates_template_info')) {
          obj['certificates_template_info'] = _CertificateTemplateInfo["default"].constructFromObject(data['certificates_template_info']);
        }

        if (data.hasOwnProperty('classic_key_details')) {
          obj['classic_key_details'] = _ClassicKeyDetailsInfo["default"].constructFromObject(data['classic_key_details']);
        }

        if (data.hasOwnProperty('cluster_gw_url')) {
          obj['cluster_gw_url'] = _ApiClient["default"].convertToType(data['cluster_gw_url'], 'String');
        }

        if (data.hasOwnProperty('display_metadata')) {
          obj['display_metadata'] = _ApiClient["default"].convertToType(data['display_metadata'], 'String');
        }

        if (data.hasOwnProperty('dynamic_secret_producer_details')) {
          obj['dynamic_secret_producer_details'] = _DynamicSecretProducerInfo["default"].constructFromObject(data['dynamic_secret_producer_details']);
        }

        if (data.hasOwnProperty('importer_info')) {
          obj['importer_info'] = _ImporterInfo["default"].constructFromObject(data['importer_info']);
        }

        if (data.hasOwnProperty('oidc_client_info')) {
          obj['oidc_client_info'] = _OidcClientInfo["default"].constructFromObject(data['oidc_client_info']);
        }

        if (data.hasOwnProperty('password_policy')) {
          obj['password_policy'] = _PasswordPolicyInfo["default"].constructFromObject(data['password_policy']);
        }

        if (data.hasOwnProperty('rotated_secret_details')) {
          obj['rotated_secret_details'] = _RotatedSecretDetailsInfo["default"].constructFromObject(data['rotated_secret_details']);
        }

        if (data.hasOwnProperty('secure_remote_access_details')) {
          obj['secure_remote_access_details'] = _SecureRemoteAccess["default"].constructFromObject(data['secure_remote_access_details']);
        }

        if (data.hasOwnProperty('static_secret_info')) {
          obj['static_secret_info'] = _StaticSecretDetailsInfo["default"].constructFromObject(data['static_secret_info']);
        }

        if (data.hasOwnProperty('tokenizer_info')) {
          obj['tokenizer_info'] = _TokenizerInfo["default"].constructFromObject(data['tokenizer_info']);
        }
      }

      return obj;
    }
  }]);

  return ItemGeneralInfo;
}();
/**
 * @member {module:model/CertificateIssueInfo} cert_issue_details
 */


ItemGeneralInfo.prototype['cert_issue_details'] = undefined;
/**
 * @member {module:model/CertificateChainInfo} certificate_chain_info
 */

ItemGeneralInfo.prototype['certificate_chain_info'] = undefined;
/**
 * @member {module:model/CertificateTemplateInfo} certificates_template_info
 */

ItemGeneralInfo.prototype['certificates_template_info'] = undefined;
/**
 * @member {module:model/ClassicKeyDetailsInfo} classic_key_details
 */

ItemGeneralInfo.prototype['classic_key_details'] = undefined;
/**
 * @member {String} cluster_gw_url
 */

ItemGeneralInfo.prototype['cluster_gw_url'] = undefined;
/**
 * @member {String} display_metadata
 */

ItemGeneralInfo.prototype['display_metadata'] = undefined;
/**
 * @member {module:model/DynamicSecretProducerInfo} dynamic_secret_producer_details
 */

ItemGeneralInfo.prototype['dynamic_secret_producer_details'] = undefined;
/**
 * @member {module:model/ImporterInfo} importer_info
 */

ItemGeneralInfo.prototype['importer_info'] = undefined;
/**
 * @member {module:model/OidcClientInfo} oidc_client_info
 */

ItemGeneralInfo.prototype['oidc_client_info'] = undefined;
/**
 * @member {module:model/PasswordPolicyInfo} password_policy
 */

ItemGeneralInfo.prototype['password_policy'] = undefined;
/**
 * @member {module:model/RotatedSecretDetailsInfo} rotated_secret_details
 */

ItemGeneralInfo.prototype['rotated_secret_details'] = undefined;
/**
 * @member {module:model/SecureRemoteAccess} secure_remote_access_details
 */

ItemGeneralInfo.prototype['secure_remote_access_details'] = undefined;
/**
 * @member {module:model/StaticSecretDetailsInfo} static_secret_info
 */

ItemGeneralInfo.prototype['static_secret_info'] = undefined;
/**
 * @member {module:model/TokenizerInfo} tokenizer_info
 */

ItemGeneralInfo.prototype['tokenizer_info'] = undefined;
var _default = ItemGeneralInfo;
exports["default"] = _default;
