// akeyless/dist/model/LogstashLogForwardingConfig.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The LogstashLogForwardingConfig model module.
 * @module model/LogstashLogForwardingConfig
 * @version 3.3.12
 */
var LogstashLogForwardingConfig = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>LogstashLogForwardingConfig</code>.
   * @alias module:model/LogstashLogForwardingConfig
   */
  function LogstashLogForwardingConfig() {
    _classCallCheck(this, LogstashLogForwardingConfig);

    LogstashLogForwardingConfig.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(LogstashLogForwardingConfig, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>LogstashLogForwardingConfig</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/LogstashLogForwardingConfig} obj Optional instance to populate.
     * @return {module:model/LogstashLogForwardingConfig} The populated <code>LogstashLogForwardingConfig</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new LogstashLogForwardingConfig();

        if (data.hasOwnProperty('logstash_dns')) {
          obj['logstash_dns'] = _ApiClient["default"].convertToType(data['logstash_dns'], 'String');
        }

        if (data.hasOwnProperty('logstash_enable_tls')) {
          obj['logstash_enable_tls'] = _ApiClient["default"].convertToType(data['logstash_enable_tls'], 'Boolean');
        }

        if (data.hasOwnProperty('logstash_protocol')) {
          obj['logstash_protocol'] = _ApiClient["default"].convertToType(data['logstash_protocol'], 'String');
        }

        if (data.hasOwnProperty('logstash_tls_certificate')) {
          obj['logstash_tls_certificate'] = _ApiClient["default"].convertToType(data['logstash_tls_certificate'], 'String');
        }
      }

      return obj;
    }
  }]);

  return LogstashLogForwardingConfig;
}();
/**
 * @member {String} logstash_dns
 */


LogstashLogForwardingConfig.prototype['logstash_dns'] = undefined;
/**
 * @member {Boolean} logstash_enable_tls
 */

LogstashLogForwardingConfig.prototype['logstash_enable_tls'] = undefined;
/**
 * @member {String} logstash_protocol
 */

LogstashLogForwardingConfig.prototype['logstash_protocol'] = undefined;
/**
 * @member {String} logstash_tls_certificate
 */

LogstashLogForwardingConfig.prototype['logstash_tls_certificate'] = undefined;
var _default = LogstashLogForwardingConfig;
exports["default"] = _default;
