// akeyless/dist/model/NativeK8sTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The NativeK8sTargetDetails model module.
 * @module model/NativeK8sTargetDetails
 * @version 3.3.16
 */
var NativeK8sTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>NativeK8sTargetDetails</code>.
   * @alias module:model/NativeK8sTargetDetails
   */
  function NativeK8sTargetDetails() {
    _classCallCheck(this, NativeK8sTargetDetails);

    NativeK8sTargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(NativeK8sTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>NativeK8sTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/NativeK8sTargetDetails} obj Optional instance to populate.
     * @return {module:model/NativeK8sTargetDetails} The populated <code>NativeK8sTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new NativeK8sTargetDetails();

        if (data.hasOwnProperty('k8s_bearer_token')) {
          obj['k8s_bearer_token'] = _ApiClient["default"].convertToType(data['k8s_bearer_token'], 'String');
        }

        if (data.hasOwnProperty('k8s_cluster_ca_certificate')) {
          obj['k8s_cluster_ca_certificate'] = _ApiClient["default"].convertToType(data['k8s_cluster_ca_certificate'], 'String');
        }

        if (data.hasOwnProperty('k8s_cluster_endpoint')) {
          obj['k8s_cluster_endpoint'] = _ApiClient["default"].convertToType(data['k8s_cluster_endpoint'], 'String');
        }

        if (data.hasOwnProperty('use_gw_service_account')) {
          obj['use_gw_service_account'] = _ApiClient["default"].convertToType(data['use_gw_service_account'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return NativeK8sTargetDetails;
}();
/**
 * @member {String} k8s_bearer_token
 */


NativeK8sTargetDetails.prototype['k8s_bearer_token'] = undefined;
/**
 * @member {String} k8s_cluster_ca_certificate
 */

NativeK8sTargetDetails.prototype['k8s_cluster_ca_certificate'] = undefined;
/**
 * @member {String} k8s_cluster_endpoint
 */

NativeK8sTargetDetails.prototype['k8s_cluster_endpoint'] = undefined;
/**
 * @member {Boolean} use_gw_service_account
 */

NativeK8sTargetDetails.prototype['use_gw_service_account'] = undefined;
var _default = NativeK8sTargetDetails;
exports["default"] = _default;
