// akeyless/dist/model/NotiForwarder.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _EmailEntry = _interopRequireDefault(require("./EmailEntry"));

var _ItemVersion = _interopRequireDefault(require("./ItemVersion"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The NotiForwarder model module.
 * @module model/NotiForwarder
 * @version 3.3.12
 */
var NotiForwarder = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>NotiForwarder</code>.
   * @alias module:model/NotiForwarder
   */
  function NotiForwarder() {
    _classCallCheck(this, NotiForwarder);

    NotiForwarder.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(NotiForwarder, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>NotiForwarder</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/NotiForwarder} obj Optional instance to populate.
     * @return {module:model/NotiForwarder} The populated <code>NotiForwarder</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new NotiForwarder();

        if (data.hasOwnProperty('client_permissions')) {
          obj['client_permissions'] = _ApiClient["default"].convertToType(data['client_permissions'], ['String']);
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('creation_date')) {
          obj['creation_date'] = _ApiClient["default"].convertToType(data['creation_date'], 'Date');
        }

        if (data.hasOwnProperty('endpoint')) {
          obj['endpoint'] = _ApiClient["default"].convertToType(data['endpoint'], 'String');
        }

        if (data.hasOwnProperty('event_types')) {
          obj['event_types'] = _ApiClient["default"].convertToType(data['event_types'], ['String']);
        }

        if (data.hasOwnProperty('gateway_cluster_id')) {
          obj['gateway_cluster_id'] = _ApiClient["default"].convertToType(data['gateway_cluster_id'], 'Number');
        }

        if (data.hasOwnProperty('is_enabled')) {
          obj['is_enabled'] = _ApiClient["default"].convertToType(data['is_enabled'], 'Boolean');
        }

        if (data.hasOwnProperty('last_version')) {
          obj['last_version'] = _ApiClient["default"].convertToType(data['last_version'], 'Number');
        }

        if (data.hasOwnProperty('modification_date')) {
          obj['modification_date'] = _ApiClient["default"].convertToType(data['modification_date'], 'Date');
        }

        if (data.hasOwnProperty('noti_forwarder_id')) {
          obj['noti_forwarder_id'] = _ApiClient["default"].convertToType(data['noti_forwarder_id'], 'Number');
        }

        if (data.hasOwnProperty('noti_forwarder_name')) {
          obj['noti_forwarder_name'] = _ApiClient["default"].convertToType(data['noti_forwarder_name'], 'String');
        }

        if (data.hasOwnProperty('noti_forwarder_type')) {
          obj['noti_forwarder_type'] = _ApiClient["default"].convertToType(data['noti_forwarder_type'], 'String');
        }

        if (data.hasOwnProperty('noti_forwarder_versions')) {
          obj['noti_forwarder_versions'] = _ApiClient["default"].convertToType(data['noti_forwarder_versions'], [_ItemVersion["default"]]);
        }

        if (data.hasOwnProperty('paths')) {
          obj['paths'] = _ApiClient["default"].convertToType(data['paths'], ['String']);
        }

        if (data.hasOwnProperty('protection_key')) {
          obj['protection_key'] = _ApiClient["default"].convertToType(data['protection_key'], 'String');
        }

        if (data.hasOwnProperty('runner_type')) {
          obj['runner_type'] = _ApiClient["default"].convertToType(data['runner_type'], 'String');
        }

        if (data.hasOwnProperty('timespan_in_seconds')) {
          obj['timespan_in_seconds'] = _ApiClient["default"].convertToType(data['timespan_in_seconds'], 'Number');
        }

        if (data.hasOwnProperty('to_emails')) {
          obj['to_emails'] = _ApiClient["default"].convertToType(data['to_emails'], [_EmailEntry["default"]]);
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }

        if (data.hasOwnProperty('with_customer_fragment')) {
          obj['with_customer_fragment'] = _ApiClient["default"].convertToType(data['with_customer_fragment'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return NotiForwarder;
}();
/**
 * @member {Array.<String>} client_permissions
 */


NotiForwarder.prototype['client_permissions'] = undefined;
/**
 * @member {String} comment
 */

NotiForwarder.prototype['comment'] = undefined;
/**
 * @member {Date} creation_date
 */

NotiForwarder.prototype['creation_date'] = undefined;
/**
 * @member {String} endpoint
 */

NotiForwarder.prototype['endpoint'] = undefined;
/**
 * @member {Array.<String>} event_types
 */

NotiForwarder.prototype['event_types'] = undefined;
/**
 * @member {Number} gateway_cluster_id
 */

NotiForwarder.prototype['gateway_cluster_id'] = undefined;
/**
 * @member {Boolean} is_enabled
 */

NotiForwarder.prototype['is_enabled'] = undefined;
/**
 * @member {Number} last_version
 */

NotiForwarder.prototype['last_version'] = undefined;
/**
 * @member {Date} modification_date
 */

NotiForwarder.prototype['modification_date'] = undefined;
/**
 * @member {Number} noti_forwarder_id
 */

NotiForwarder.prototype['noti_forwarder_id'] = undefined;
/**
 * @member {String} noti_forwarder_name
 */

NotiForwarder.prototype['noti_forwarder_name'] = undefined;
/**
 * @member {String} noti_forwarder_type
 */

NotiForwarder.prototype['noti_forwarder_type'] = undefined;
/**
 * @member {Array.<module:model/ItemVersion>} noti_forwarder_versions
 */

NotiForwarder.prototype['noti_forwarder_versions'] = undefined;
/**
 * @member {Array.<String>} paths
 */

NotiForwarder.prototype['paths'] = undefined;
/**
 * @member {String} protection_key
 */

NotiForwarder.prototype['protection_key'] = undefined;
/**
 * @member {String} runner_type
 */

NotiForwarder.prototype['runner_type'] = undefined;
/**
 * @member {Number} timespan_in_seconds
 */

NotiForwarder.prototype['timespan_in_seconds'] = undefined;
/**
 * @member {Array.<module:model/EmailEntry>} to_emails
 */

NotiForwarder.prototype['to_emails'] = undefined;
/**
 * @member {String} username
 */

NotiForwarder.prototype['username'] = undefined;
/**
 * @member {Boolean} with_customer_fragment
 */

NotiForwarder.prototype['with_customer_fragment'] = undefined;
var _default = NotiForwarder;
exports["default"] = _default;
