// akeyless/dist/model/OidcClientInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AccessPermissionAssignment = _interopRequireDefault(require("./AccessPermissionAssignment"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The OidcClientInfo model module.
 * @module model/OidcClientInfo
 * @version 3.3.16
 */
var OidcClientInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OidcClientInfo</code>.
   * @alias module:model/OidcClientInfo
   */
  function OidcClientInfo() {
    _classCallCheck(this, OidcClientInfo);

    OidcClientInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(OidcClientInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OidcClientInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OidcClientInfo} obj Optional instance to populate.
     * @return {module:model/OidcClientInfo} The populated <code>OidcClientInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OidcClientInfo();

        if (data.hasOwnProperty('access_permission_assignment')) {
          obj['access_permission_assignment'] = _ApiClient["default"].convertToType(data['access_permission_assignment'], [_AccessPermissionAssignment["default"]]);
        }

        if (data.hasOwnProperty('audience')) {
          obj['audience'] = _ApiClient["default"].convertToType(data['audience'], ['String']);
        }

        if (data.hasOwnProperty('client_id')) {
          obj['client_id'] = _ApiClient["default"].convertToType(data['client_id'], 'String');
        }

        if (data.hasOwnProperty('grant_types')) {
          obj['grant_types'] = _ApiClient["default"].convertToType(data['grant_types'], ['String']);
        }

        if (data.hasOwnProperty('logout_uris')) {
          obj['logout_uris'] = _ApiClient["default"].convertToType(data['logout_uris'], ['String']);
        }

        if (data.hasOwnProperty('public')) {
          obj['public'] = _ApiClient["default"].convertToType(data['public'], 'Boolean');
        }

        if (data.hasOwnProperty('redirect_uris')) {
          obj['redirect_uris'] = _ApiClient["default"].convertToType(data['redirect_uris'], ['String']);
        }

        if (data.hasOwnProperty('response_types')) {
          obj['response_types'] = _ApiClient["default"].convertToType(data['response_types'], ['String']);
        }

        if (data.hasOwnProperty('scopes')) {
          obj['scopes'] = _ApiClient["default"].convertToType(data['scopes'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return OidcClientInfo;
}();
/**
 * @member {Array.<module:model/AccessPermissionAssignment>} access_permission_assignment
 */


OidcClientInfo.prototype['access_permission_assignment'] = undefined;
/**
 * @member {Array.<String>} audience
 */

OidcClientInfo.prototype['audience'] = undefined;
/**
 * @member {String} client_id
 */

OidcClientInfo.prototype['client_id'] = undefined;
/**
 * @member {Array.<String>} grant_types
 */

OidcClientInfo.prototype['grant_types'] = undefined;
/**
 * @member {Array.<String>} logout_uris
 */

OidcClientInfo.prototype['logout_uris'] = undefined;
/**
 * @member {Boolean} public
 */

OidcClientInfo.prototype['public'] = undefined;
/**
 * @member {Array.<String>} redirect_uris
 */

OidcClientInfo.prototype['redirect_uris'] = undefined;
/**
 * @member {Array.<String>} response_types
 */

OidcClientInfo.prototype['response_types'] = undefined;
/**
 * @member {Array.<String>} scopes
 */

OidcClientInfo.prototype['scopes'] = undefined;
var _default = OidcClientInfo;
exports["default"] = _default;
