// akeyless/dist/model/RabbitMQTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The RabbitMQTargetDetails model module.
 * @module model/RabbitMQTargetDetails
 * @version 3.5.0
 */
var RabbitMQTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RabbitMQTargetDetails</code>.
   * @alias module:model/RabbitMQTargetDetails
   */
  function RabbitMQTargetDetails() {
    _classCallCheck(this, RabbitMQTargetDetails);

    RabbitMQTargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RabbitMQTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>RabbitMQTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RabbitMQTargetDetails} obj Optional instance to populate.
     * @return {module:model/RabbitMQTargetDetails} The populated <code>RabbitMQTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RabbitMQTargetDetails();

        if (data.hasOwnProperty('rabbitmq_server_password')) {
          obj['rabbitmq_server_password'] = _ApiClient["default"].convertToType(data['rabbitmq_server_password'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq_server_uri')) {
          obj['rabbitmq_server_uri'] = _ApiClient["default"].convertToType(data['rabbitmq_server_uri'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq_server_user')) {
          obj['rabbitmq_server_user'] = _ApiClient["default"].convertToType(data['rabbitmq_server_user'], 'String');
        }
      }

      return obj;
    }
  }]);

  return RabbitMQTargetDetails;
}();
/**
 * @member {String} rabbitmq_server_password
 */


RabbitMQTargetDetails.prototype['rabbitmq_server_password'] = undefined;
/**
 * @member {String} rabbitmq_server_uri
 */

RabbitMQTargetDetails.prototype['rabbitmq_server_uri'] = undefined;
/**
 * @member {String} rabbitmq_server_user
 */

RabbitMQTargetDetails.prototype['rabbitmq_server_user'] = undefined;
var _default = RabbitMQTargetDetails;
exports["default"] = _default;
