// akeyless/dist/model/RequestAccess.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The RequestAccess model module.
 * @module model/RequestAccess
 * @version 3.3.9
 */
var RequestAccess = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RequestAccess</code>.
   * @alias module:model/RequestAccess
   * @param capability {Array.<String>} List of the required capabilities options: [read, update, delete]
   * @param name {String} Item type
   */
  function RequestAccess(capability, name) {
    _classCallCheck(this, RequestAccess);

    RequestAccess.initialize(this, capability, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RequestAccess, null, [{
    key: "initialize",
    value: function initialize(obj, capability, name) {
      obj['capability'] = capability;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>RequestAccess</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RequestAccess} obj Optional instance to populate.
     * @return {module:model/RequestAccess} The populated <code>RequestAccess</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RequestAccess();

        if (data.hasOwnProperty('capability')) {
          obj['capability'] = _ApiClient["default"].convertToType(data['capability'], ['String']);
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return RequestAccess;
}();
/**
 * List of the required capabilities options: [read, update, delete]
 * @member {Array.<String>} capability
 */


RequestAccess.prototype['capability'] = undefined;
/**
 * Deprecated - use description
 * @member {String} comment
 */

RequestAccess.prototype['comment'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

RequestAccess.prototype['description'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

RequestAccess.prototype['json'] = false;
/**
 * Item type
 * @member {String} name
 */

RequestAccess.prototype['name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

RequestAccess.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

RequestAccess.prototype['uid-token'] = undefined;
var _default = RequestAccess;
exports["default"] = _default;
