// akeyless/dist/model/SalesforceTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SalesforceTargetDetails model module.
 * @module model/SalesforceTargetDetails
 * @version 3.5.0
 */
var SalesforceTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SalesforceTargetDetails</code>.
   * SalesforceTargetDetails
   * @alias module:model/SalesforceTargetDetails
   */
  function SalesforceTargetDetails() {
    _classCallCheck(this, SalesforceTargetDetails);

    SalesforceTargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SalesforceTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SalesforceTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SalesforceTargetDetails} obj Optional instance to populate.
     * @return {module:model/SalesforceTargetDetails} The populated <code>SalesforceTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SalesforceTargetDetails();

        if (data.hasOwnProperty('app_private_key')) {
          obj['app_private_key'] = _ApiClient["default"].convertToType(data['app_private_key'], ['Number']);
        }

        if (data.hasOwnProperty('auth_flow')) {
          obj['auth_flow'] = _ApiClient["default"].convertToType(data['auth_flow'], 'String');
        }

        if (data.hasOwnProperty('ca_cert_data')) {
          obj['ca_cert_data'] = _ApiClient["default"].convertToType(data['ca_cert_data'], ['Number']);
        }

        if (data.hasOwnProperty('ca_cert_name')) {
          obj['ca_cert_name'] = _ApiClient["default"].convertToType(data['ca_cert_name'], 'String');
        }

        if (data.hasOwnProperty('client_id')) {
          obj['client_id'] = _ApiClient["default"].convertToType(data['client_id'], 'String');
        }

        if (data.hasOwnProperty('client_secret')) {
          obj['client_secret'] = _ApiClient["default"].convertToType(data['client_secret'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('security_token')) {
          obj['security_token'] = _ApiClient["default"].convertToType(data['security_token'], 'String');
        }

        if (data.hasOwnProperty('tenant_url')) {
          obj['tenant_url'] = _ApiClient["default"].convertToType(data['tenant_url'], 'String');
        }

        if (data.hasOwnProperty('user_name')) {
          obj['user_name'] = _ApiClient["default"].convertToType(data['user_name'], 'String');
        }
      }

      return obj;
    }
  }]);

  return SalesforceTargetDetails;
}();
/**
 * params needed for jwt auth AppPrivateKey is the rsa private key in PEM format
 * @member {Array.<Number>} app_private_key
 */


SalesforceTargetDetails.prototype['app_private_key'] = undefined;
/**
 * @member {String} auth_flow
 */

SalesforceTargetDetails.prototype['auth_flow'] = undefined;
/**
 * CACertData is the rsa 4096 certificate data in PEM format
 * @member {Array.<Number>} ca_cert_data
 */

SalesforceTargetDetails.prototype['ca_cert_data'] = undefined;
/**
 * CACertName is the name of the certificate in SalesForce tenant
 * @member {String} ca_cert_name
 */

SalesforceTargetDetails.prototype['ca_cert_name'] = undefined;
/**
 * @member {String} client_id
 */

SalesforceTargetDetails.prototype['client_id'] = undefined;
/**
 * params needed for password auth
 * @member {String} client_secret
 */

SalesforceTargetDetails.prototype['client_secret'] = undefined;
/**
 * @member {String} password
 */

SalesforceTargetDetails.prototype['password'] = undefined;
/**
 * @member {String} security_token
 */

SalesforceTargetDetails.prototype['security_token'] = undefined;
/**
 * @member {String} tenant_url
 */

SalesforceTargetDetails.prototype['tenant_url'] = undefined;
/**
 * @member {String} user_name
 */

SalesforceTargetDetails.prototype['user_name'] = undefined;
var _default = SalesforceTargetDetails;
exports["default"] = _default;
