// akeyless/dist/model/SecretInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SecretInfo model module.
 * @module model/SecretInfo
 * @version 3.3.14
 */
var SecretInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SecretInfo</code>.
   * @alias module:model/SecretInfo
   */
  function SecretInfo() {
    _classCallCheck(this, SecretInfo);

    SecretInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SecretInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SecretInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SecretInfo} obj Optional instance to populate.
     * @return {module:model/SecretInfo} The populated <code>SecretInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SecretInfo();

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('expiration')) {
          obj['expiration'] = _ApiClient["default"].convertToType(data['expiration'], 'Date');
        }

        if (data.hasOwnProperty('last_retrieved')) {
          obj['last_retrieved'] = _ApiClient["default"].convertToType(data['last_retrieved'], 'Date');
        }

        if (data.hasOwnProperty('location')) {
          obj['location'] = _ApiClient["default"].convertToType(data['location'], Object);
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('secret_id')) {
          obj['secret_id'] = _ApiClient["default"].convertToType(data['secret_id'], 'String');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], 'Boolean');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
      }

      return obj;
    }
  }]);

  return SecretInfo;
}();
/**
 * @member {Date} created
 */


SecretInfo.prototype['created'] = undefined;
/**
 * @member {String} description
 */

SecretInfo.prototype['description'] = undefined;
/**
 * @member {Date} expiration
 */

SecretInfo.prototype['expiration'] = undefined;
/**
 * @member {Date} last_retrieved
 */

SecretInfo.prototype['last_retrieved'] = undefined;
/**
 * @member {Object} location
 */

SecretInfo.prototype['location'] = undefined;
/**
 * @member {String} name
 */

SecretInfo.prototype['name'] = undefined;
/**
 * @member {String} secret_id
 */

SecretInfo.prototype['secret_id'] = undefined;
/**
 * @member {Boolean} status
 */

SecretInfo.prototype['status'] = undefined;
/**
 * @member {Object.<String, String>} tags
 */

SecretInfo.prototype['tags'] = undefined;
/**
 * @member {String} type
 */

SecretInfo.prototype['type'] = undefined;
var _default = SecretInfo;
exports["default"] = _default;
