// akeyless/dist/model/ServerInventoryPayload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ServerInventoryPayload model module.
 * @module model/ServerInventoryPayload
 * @version 3.5.0
 */
var ServerInventoryPayload = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ServerInventoryPayload</code>.
   * @alias module:model/ServerInventoryPayload
   */
  function ServerInventoryPayload() {
    _classCallCheck(this, ServerInventoryPayload);

    ServerInventoryPayload.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ServerInventoryPayload, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ServerInventoryPayload</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ServerInventoryPayload} obj Optional instance to populate.
     * @return {module:model/ServerInventoryPayload} The populated <code>ServerInventoryPayload</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ServerInventoryPayload();

        if (data.hasOwnProperty('auto_rotate')) {
          obj['auto_rotate'] = _ApiClient["default"].convertToType(data['auto_rotate'], 'Boolean');
        }

        if (data.hasOwnProperty('auto_rotate_interval_in_days')) {
          obj['auto_rotate_interval_in_days'] = _ApiClient["default"].convertToType(data['auto_rotate_interval_in_days'], 'Number');
        }

        if (data.hasOwnProperty('auto_rotate_rotation_hour')) {
          obj['auto_rotate_rotation_hour'] = _ApiClient["default"].convertToType(data['auto_rotate_rotation_hour'], 'Number');
        }

        if (data.hasOwnProperty('enable_rdp_sra')) {
          obj['enable_rdp_sra'] = _ApiClient["default"].convertToType(data['enable_rdp_sra'], 'Boolean');
        }

        if (data.hasOwnProperty('migration_target_id')) {
          obj['migration_target_id'] = _ApiClient["default"].convertToType(data['migration_target_id'], 'Number');
        }

        if (data.hasOwnProperty('server_targets_path_template')) {
          obj['server_targets_path_template'] = _ApiClient["default"].convertToType(data['server_targets_path_template'], 'String');
        }

        if (data.hasOwnProperty('users_ignore_list')) {
          obj['users_ignore_list'] = _ApiClient["default"].convertToType(data['users_ignore_list'], {
            'String': 'Boolean'
          });
        }

        if (data.hasOwnProperty('users_rotated_secrets_path_template')) {
          obj['users_rotated_secrets_path_template'] = _ApiClient["default"].convertToType(data['users_rotated_secrets_path_template'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ServerInventoryPayload;
}();
/**
 * @member {Boolean} auto_rotate
 */


ServerInventoryPayload.prototype['auto_rotate'] = undefined;
/**
 * @member {Number} auto_rotate_interval_in_days
 */

ServerInventoryPayload.prototype['auto_rotate_interval_in_days'] = undefined;
/**
 * @member {Number} auto_rotate_rotation_hour
 */

ServerInventoryPayload.prototype['auto_rotate_rotation_hour'] = undefined;
/**
 * @member {Boolean} enable_rdp_sra
 */

ServerInventoryPayload.prototype['enable_rdp_sra'] = undefined;
/**
 * @member {Number} migration_target_id
 */

ServerInventoryPayload.prototype['migration_target_id'] = undefined;
/**
 * @member {String} server_targets_path_template
 */

ServerInventoryPayload.prototype['server_targets_path_template'] = undefined;
/**
 * @member {Object.<String, Boolean>} users_ignore_list
 */

ServerInventoryPayload.prototype['users_ignore_list'] = undefined;
/**
 * @member {String} users_rotated_secrets_path_template
 */

ServerInventoryPayload.prototype['users_rotated_secrets_path_template'] = undefined;
var _default = ServerInventoryPayload;
exports["default"] = _default;
