// akeyless/dist/model/SignEcDsa.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SignEcDsa model module.
 * @module model/SignEcDsa
 * @version 3.5.0
 */
var SignEcDsa = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SignEcDsa</code>.
   * signEcDsa Calculates the signature of a given message using ECDSA and a sha hash algorithm matching the key size
   * @alias module:model/SignEcDsa
   * @param message {String} The input message to sign in a base64 format
   */
  function SignEcDsa(message) {
    _classCallCheck(this, SignEcDsa);

    SignEcDsa.initialize(this, message);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SignEcDsa, null, [{
    key: "initialize",
    value: function initialize(obj, message) {
      obj['message'] = message;
    }
    /**
     * Constructs a <code>SignEcDsa</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SignEcDsa} obj Optional instance to populate.
     * @return {module:model/SignEcDsa} The populated <code>SignEcDsa</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SignEcDsa();

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('item-id')) {
          obj['item-id'] = _ApiClient["default"].convertToType(data['item-id'], 'Number');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key-name')) {
          obj['key-name'] = _ApiClient["default"].convertToType(data['key-name'], 'String');
        }

        if (data.hasOwnProperty('message')) {
          obj['message'] = _ApiClient["default"].convertToType(data['message'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return SignEcDsa;
}();
/**
 * The display id of the EC key to use for the signing process
 * @member {String} display-id
 */


SignEcDsa.prototype['display-id'] = undefined;
/**
 * The item id of the EC key to use for the signing process
 * @member {Number} item-id
 */

SignEcDsa.prototype['item-id'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

SignEcDsa.prototype['json'] = false;
/**
 * The name of the EC key to use for the signing process
 * @member {String} key-name
 */

SignEcDsa.prototype['key-name'] = undefined;
/**
 * The input message to sign in a base64 format
 * @member {String} message
 */

SignEcDsa.prototype['message'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

SignEcDsa.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

SignEcDsa.prototype['uid-token'] = undefined;
var _default = SignEcDsa;
exports["default"] = _default;
