// akeyless/dist/model/SignDataWithClassicKey.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SignDataWithClassicKey model module.
 * @module model/SignDataWithClassicKey
 * @version 3.5.0
 */
var SignDataWithClassicKey = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SignDataWithClassicKey</code>.
   * @alias module:model/SignDataWithClassicKey
   * @param data {String} Data
   * @param displayId {String} The name of the key to use in the sign data process
   * @param name {String} ClassicKey name
   * @param version {Number} classic key version
   */
  function SignDataWithClassicKey(data, displayId, name, version) {
    _classCallCheck(this, SignDataWithClassicKey);

    SignDataWithClassicKey.initialize(this, data, displayId, name, version);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SignDataWithClassicKey, null, [{
    key: "initialize",
    value: function initialize(obj, data, displayId, name, version) {
      obj['data'] = data;
      obj['display-id'] = displayId;
      obj['name'] = name;
      obj['version'] = version;
    }
    /**
     * Constructs a <code>SignDataWithClassicKey</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SignDataWithClassicKey} obj Optional instance to populate.
     * @return {module:model/SignDataWithClassicKey} The populated <code>SignDataWithClassicKey</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SignDataWithClassicKey();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], 'String');
        }

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('hashed')) {
          obj['hashed'] = _ApiClient["default"].convertToType(data['hashed'], 'Boolean');
        }

        if (data.hasOwnProperty('hashing-method')) {
          obj['hashing-method'] = _ApiClient["default"].convertToType(data['hashing-method'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return SignDataWithClassicKey;
}();
/**
 * Data
 * @member {String} data
 */


SignDataWithClassicKey.prototype['data'] = undefined;
/**
 * The name of the key to use in the sign data process
 * @member {String} display-id
 */

SignDataWithClassicKey.prototype['display-id'] = undefined;
/**
 * Defines whether the data should be hashed as part of the signing. If true, the data will not be hashed
 * @member {Boolean} hashed
 * @default false
 */

SignDataWithClassicKey.prototype['hashed'] = false;
/**
 * HashingMethod
 * @member {String} hashing-method
 * @default 'SHA256'
 */

SignDataWithClassicKey.prototype['hashing-method'] = 'SHA256';
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

SignDataWithClassicKey.prototype['json'] = false;
/**
 * ClassicKey name
 * @member {String} name
 */

SignDataWithClassicKey.prototype['name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

SignDataWithClassicKey.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

SignDataWithClassicKey.prototype['uid-token'] = undefined;
/**
 * classic key version
 * @member {Number} version
 */

SignDataWithClassicKey.prototype['version'] = undefined;
var _default = SignDataWithClassicKey;
exports["default"] = _default;
