// akeyless/dist/model/SignGPG.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SignGPG model module.
 * @module model/SignGPG
 * @version 3.3.16
 */
var SignGPG = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SignGPG</code>.
   * @alias module:model/SignGPG
   * @param keyName {String} The name of the key to use in the encryption process
   * @param message {String} The message to be signed in base64 format
   */
  function SignGPG(keyName, message) {
    _classCallCheck(this, SignGPG);

    SignGPG.initialize(this, keyName, message);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SignGPG, null, [{
    key: "initialize",
    value: function initialize(obj, keyName, message) {
      obj['key-name'] = keyName;
      obj['message'] = message;
    }
    /**
     * Constructs a <code>SignGPG</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SignGPG} obj Optional instance to populate.
     * @return {module:model/SignGPG} The populated <code>SignGPG</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SignGPG();

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('item-id')) {
          obj['item-id'] = _ApiClient["default"].convertToType(data['item-id'], 'Number');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key-name')) {
          obj['key-name'] = _ApiClient["default"].convertToType(data['key-name'], 'String');
        }

        if (data.hasOwnProperty('message')) {
          obj['message'] = _ApiClient["default"].convertToType(data['message'], 'String');
        }

        if (data.hasOwnProperty('passphrase')) {
          obj['passphrase'] = _ApiClient["default"].convertToType(data['passphrase'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return SignGPG;
}();
/**
 * The display id of the key to use in the encryption process
 * @member {String} display-id
 */


SignGPG.prototype['display-id'] = undefined;
/**
 * The item id of the key to use in the encryption process
 * @member {Number} item-id
 */

SignGPG.prototype['item-id'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

SignGPG.prototype['json'] = false;
/**
 * The name of the key to use in the encryption process
 * @member {String} key-name
 */

SignGPG.prototype['key-name'] = undefined;
/**
 * The message to be signed in base64 format
 * @member {String} message
 */

SignGPG.prototype['message'] = undefined;
/**
 * Passphrase that was used to generate the key
 * @member {String} passphrase
 */

SignGPG.prototype['passphrase'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

SignGPG.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

SignGPG.prototype['uid-token'] = undefined;
var _default = SignGPG;
exports["default"] = _default;
