// akeyless/dist/model/SSHTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SSHTargetDetails model module.
 * @module model/SSHTargetDetails
 * @version 3.3.16
 */
var SSHTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SSHTargetDetails</code>.
   * @alias module:model/SSHTargetDetails
   */
  function SSHTargetDetails() {
    _classCallCheck(this, SSHTargetDetails);

    SSHTargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SSHTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SSHTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SSHTargetDetails} obj Optional instance to populate.
     * @return {module:model/SSHTargetDetails} The populated <code>SSHTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SSHTargetDetails();

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('port')) {
          obj['port'] = _ApiClient["default"].convertToType(data['port'], 'String');
        }

        if (data.hasOwnProperty('private_key')) {
          obj['private_key'] = _ApiClient["default"].convertToType(data['private_key'], 'String');
        }

        if (data.hasOwnProperty('private_key_password')) {
          obj['private_key_password'] = _ApiClient["default"].convertToType(data['private_key_password'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return SSHTargetDetails;
}();
/**
 * @member {String} host
 */


SSHTargetDetails.prototype['host'] = undefined;
/**
 * @member {String} password
 */

SSHTargetDetails.prototype['password'] = undefined;
/**
 * @member {String} port
 */

SSHTargetDetails.prototype['port'] = undefined;
/**
 * @member {String} private_key
 */

SSHTargetDetails.prototype['private_key'] = undefined;
/**
 * @member {String} private_key_password
 */

SSHTargetDetails.prototype['private_key_password'] = undefined;
/**
 * @member {String} username
 */

SSHTargetDetails.prototype['username'] = undefined;
var _default = SSHTargetDetails;
exports["default"] = _default;
