// akeyless/dist/model/TargetTypeDetailsInput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AWSTargetDetails = _interopRequireDefault(require("./AWSTargetDetails"));

var _ArtifactoryTargetDetails = _interopRequireDefault(require("./ArtifactoryTargetDetails"));

var _AzureTargetDetails = _interopRequireDefault(require("./AzureTargetDetails"));

var _ChefTargetDetails = _interopRequireDefault(require("./ChefTargetDetails"));

var _CustomTargetDetails = _interopRequireDefault(require("./CustomTargetDetails"));

var _DbTargetDetails = _interopRequireDefault(require("./DbTargetDetails"));

var _DockerhubTargetDetails = _interopRequireDefault(require("./DockerhubTargetDetails"));

var _EKSTargetDetails = _interopRequireDefault(require("./EKSTargetDetails"));

var _GKETargetDetails = _interopRequireDefault(require("./GKETargetDetails"));

var _GcpTargetDetails = _interopRequireDefault(require("./GcpTargetDetails"));

var _GithubTargetDetails = _interopRequireDefault(require("./GithubTargetDetails"));

var _GlobalSignAtlasTargetDetails = _interopRequireDefault(require("./GlobalSignAtlasTargetDetails"));

var _GlobalSignGCCTargetDetails = _interopRequireDefault(require("./GlobalSignGCCTargetDetails"));

var _LdapTargetDetails = _interopRequireDefault(require("./LdapTargetDetails"));

var _LinkedTargetDetails = _interopRequireDefault(require("./LinkedTargetDetails"));

var _MongoDBTargetDetails = _interopRequireDefault(require("./MongoDBTargetDetails"));

var _NativeK8sTargetDetails = _interopRequireDefault(require("./NativeK8sTargetDetails"));

var _PingTargetDetails = _interopRequireDefault(require("./PingTargetDetails"));

var _RabbitMQTargetDetails = _interopRequireDefault(require("./RabbitMQTargetDetails"));

var _SSHTargetDetails = _interopRequireDefault(require("./SSHTargetDetails"));

var _SalesforceTargetDetails = _interopRequireDefault(require("./SalesforceTargetDetails"));

var _VenafiTargetDetails = _interopRequireDefault(require("./VenafiTargetDetails"));

var _WebTargetDetails = _interopRequireDefault(require("./WebTargetDetails"));

var _WindowsTargetDetails = _interopRequireDefault(require("./WindowsTargetDetails"));

var _ZeroSSLTargetDetails = _interopRequireDefault(require("./ZeroSSLTargetDetails"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetTypeDetailsInput model module.
 * @module model/TargetTypeDetailsInput
 * @version 3.5.0
 */
var TargetTypeDetailsInput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetTypeDetailsInput</code>.
   * @alias module:model/TargetTypeDetailsInput
   */
  function TargetTypeDetailsInput() {
    _classCallCheck(this, TargetTypeDetailsInput);

    TargetTypeDetailsInput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetTypeDetailsInput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>TargetTypeDetailsInput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetTypeDetailsInput} obj Optional instance to populate.
     * @return {module:model/TargetTypeDetailsInput} The populated <code>TargetTypeDetailsInput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetTypeDetailsInput();

        if (data.hasOwnProperty('artifactory_target_details')) {
          obj['artifactory_target_details'] = _ArtifactoryTargetDetails["default"].constructFromObject(data['artifactory_target_details']);
        }

        if (data.hasOwnProperty('aws_target_details')) {
          obj['aws_target_details'] = _AWSTargetDetails["default"].constructFromObject(data['aws_target_details']);
        }

        if (data.hasOwnProperty('azure_target_details')) {
          obj['azure_target_details'] = _AzureTargetDetails["default"].constructFromObject(data['azure_target_details']);
        }

        if (data.hasOwnProperty('chef_target_details')) {
          obj['chef_target_details'] = _ChefTargetDetails["default"].constructFromObject(data['chef_target_details']);
        }

        if (data.hasOwnProperty('custom_target_details')) {
          obj['custom_target_details'] = _CustomTargetDetails["default"].constructFromObject(data['custom_target_details']);
        }

        if (data.hasOwnProperty('db_target_details')) {
          obj['db_target_details'] = _DbTargetDetails["default"].constructFromObject(data['db_target_details']);
        }

        if (data.hasOwnProperty('dockerhub_target_details')) {
          obj['dockerhub_target_details'] = _DockerhubTargetDetails["default"].constructFromObject(data['dockerhub_target_details']);
        }

        if (data.hasOwnProperty('eks_target_details')) {
          obj['eks_target_details'] = _EKSTargetDetails["default"].constructFromObject(data['eks_target_details']);
        }

        if (data.hasOwnProperty('gcp_target_details')) {
          obj['gcp_target_details'] = _GcpTargetDetails["default"].constructFromObject(data['gcp_target_details']);
        }

        if (data.hasOwnProperty('github_target_details')) {
          obj['github_target_details'] = _GithubTargetDetails["default"].constructFromObject(data['github_target_details']);
        }

        if (data.hasOwnProperty('gke_target_details')) {
          obj['gke_target_details'] = _GKETargetDetails["default"].constructFromObject(data['gke_target_details']);
        }

        if (data.hasOwnProperty('globalsign_atlas_target_details')) {
          obj['globalsign_atlas_target_details'] = _GlobalSignAtlasTargetDetails["default"].constructFromObject(data['globalsign_atlas_target_details']);
        }

        if (data.hasOwnProperty('globalsign_target_details')) {
          obj['globalsign_target_details'] = _GlobalSignGCCTargetDetails["default"].constructFromObject(data['globalsign_target_details']);
        }

        if (data.hasOwnProperty('ldap_target_details')) {
          obj['ldap_target_details'] = _LdapTargetDetails["default"].constructFromObject(data['ldap_target_details']);
        }

        if (data.hasOwnProperty('linked_target_details')) {
          obj['linked_target_details'] = _LinkedTargetDetails["default"].constructFromObject(data['linked_target_details']);
        }

        if (data.hasOwnProperty('mongo_db_target_details')) {
          obj['mongo_db_target_details'] = _MongoDBTargetDetails["default"].constructFromObject(data['mongo_db_target_details']);
        }

        if (data.hasOwnProperty('native_k8s_target_details')) {
          obj['native_k8s_target_details'] = _NativeK8sTargetDetails["default"].constructFromObject(data['native_k8s_target_details']);
        }

        if (data.hasOwnProperty('ping_target_details')) {
          obj['ping_target_details'] = _PingTargetDetails["default"].constructFromObject(data['ping_target_details']);
        }

        if (data.hasOwnProperty('rabbit_mq_target_details')) {
          obj['rabbit_mq_target_details'] = _RabbitMQTargetDetails["default"].constructFromObject(data['rabbit_mq_target_details']);
        }

        if (data.hasOwnProperty('salesforce_target_details')) {
          obj['salesforce_target_details'] = _SalesforceTargetDetails["default"].constructFromObject(data['salesforce_target_details']);
        }

        if (data.hasOwnProperty('ssh_target_details')) {
          obj['ssh_target_details'] = _SSHTargetDetails["default"].constructFromObject(data['ssh_target_details']);
        }

        if (data.hasOwnProperty('venafi_target_details')) {
          obj['venafi_target_details'] = _VenafiTargetDetails["default"].constructFromObject(data['venafi_target_details']);
        }

        if (data.hasOwnProperty('web_target_details')) {
          obj['web_target_details'] = _WebTargetDetails["default"].constructFromObject(data['web_target_details']);
        }

        if (data.hasOwnProperty('windows_target_details')) {
          obj['windows_target_details'] = _WindowsTargetDetails["default"].constructFromObject(data['windows_target_details']);
        }

        if (data.hasOwnProperty('zerossl_target_details')) {
          obj['zerossl_target_details'] = _ZeroSSLTargetDetails["default"].constructFromObject(data['zerossl_target_details']);
        }
      }

      return obj;
    }
  }]);

  return TargetTypeDetailsInput;
}();
/**
 * @member {module:model/ArtifactoryTargetDetails} artifactory_target_details
 */


TargetTypeDetailsInput.prototype['artifactory_target_details'] = undefined;
/**
 * @member {module:model/AWSTargetDetails} aws_target_details
 */

TargetTypeDetailsInput.prototype['aws_target_details'] = undefined;
/**
 * @member {module:model/AzureTargetDetails} azure_target_details
 */

TargetTypeDetailsInput.prototype['azure_target_details'] = undefined;
/**
 * @member {module:model/ChefTargetDetails} chef_target_details
 */

TargetTypeDetailsInput.prototype['chef_target_details'] = undefined;
/**
 * @member {module:model/CustomTargetDetails} custom_target_details
 */

TargetTypeDetailsInput.prototype['custom_target_details'] = undefined;
/**
 * @member {module:model/DbTargetDetails} db_target_details
 */

TargetTypeDetailsInput.prototype['db_target_details'] = undefined;
/**
 * @member {module:model/DockerhubTargetDetails} dockerhub_target_details
 */

TargetTypeDetailsInput.prototype['dockerhub_target_details'] = undefined;
/**
 * @member {module:model/EKSTargetDetails} eks_target_details
 */

TargetTypeDetailsInput.prototype['eks_target_details'] = undefined;
/**
 * @member {module:model/GcpTargetDetails} gcp_target_details
 */

TargetTypeDetailsInput.prototype['gcp_target_details'] = undefined;
/**
 * @member {module:model/GithubTargetDetails} github_target_details
 */

TargetTypeDetailsInput.prototype['github_target_details'] = undefined;
/**
 * @member {module:model/GKETargetDetails} gke_target_details
 */

TargetTypeDetailsInput.prototype['gke_target_details'] = undefined;
/**
 * @member {module:model/GlobalSignAtlasTargetDetails} globalsign_atlas_target_details
 */

TargetTypeDetailsInput.prototype['globalsign_atlas_target_details'] = undefined;
/**
 * @member {module:model/GlobalSignGCCTargetDetails} globalsign_target_details
 */

TargetTypeDetailsInput.prototype['globalsign_target_details'] = undefined;
/**
 * @member {module:model/LdapTargetDetails} ldap_target_details
 */

TargetTypeDetailsInput.prototype['ldap_target_details'] = undefined;
/**
 * @member {module:model/LinkedTargetDetails} linked_target_details
 */

TargetTypeDetailsInput.prototype['linked_target_details'] = undefined;
/**
 * @member {module:model/MongoDBTargetDetails} mongo_db_target_details
 */

TargetTypeDetailsInput.prototype['mongo_db_target_details'] = undefined;
/**
 * @member {module:model/NativeK8sTargetDetails} native_k8s_target_details
 */

TargetTypeDetailsInput.prototype['native_k8s_target_details'] = undefined;
/**
 * @member {module:model/PingTargetDetails} ping_target_details
 */

TargetTypeDetailsInput.prototype['ping_target_details'] = undefined;
/**
 * @member {module:model/RabbitMQTargetDetails} rabbit_mq_target_details
 */

TargetTypeDetailsInput.prototype['rabbit_mq_target_details'] = undefined;
/**
 * @member {module:model/SalesforceTargetDetails} salesforce_target_details
 */

TargetTypeDetailsInput.prototype['salesforce_target_details'] = undefined;
/**
 * @member {module:model/SSHTargetDetails} ssh_target_details
 */

TargetTypeDetailsInput.prototype['ssh_target_details'] = undefined;
/**
 * @member {module:model/VenafiTargetDetails} venafi_target_details
 */

TargetTypeDetailsInput.prototype['venafi_target_details'] = undefined;
/**
 * @member {module:model/WebTargetDetails} web_target_details
 */

TargetTypeDetailsInput.prototype['web_target_details'] = undefined;
/**
 * @member {module:model/WindowsTargetDetails} windows_target_details
 */

TargetTypeDetailsInput.prototype['windows_target_details'] = undefined;
/**
 * @member {module:model/ZeroSSLTargetDetails} zerossl_target_details
 */

TargetTypeDetailsInput.prototype['zerossl_target_details'] = undefined;
var _default = TargetTypeDetailsInput;
exports["default"] = _default;
