// akeyless/dist/model/UpdateZeroSSLTarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateZeroSSLTarget model module.
 * @module model/UpdateZeroSSLTarget
 * @version 3.3.16
 */
var UpdateZeroSSLTarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateZeroSSLTarget</code>.
   * @alias module:model/UpdateZeroSSLTarget
   * @param apiKey {String} API Key of the ZeroSSLTarget account
   * @param imapFqdn {String} ImapFQDN of the IMAP service, FQDN or IPv4 address. Must be FQDN if the IMAP is using TLS
   * @param imapPassword {String} ImapPassword to access the IMAP service
   * @param imapUsername {String} ImapUsername to access the IMAP service
   * @param name {String} Target name
   */
  function UpdateZeroSSLTarget(apiKey, imapFqdn, imapPassword, imapUsername, name) {
    _classCallCheck(this, UpdateZeroSSLTarget);

    UpdateZeroSSLTarget.initialize(this, apiKey, imapFqdn, imapPassword, imapUsername, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateZeroSSLTarget, null, [{
    key: "initialize",
    value: function initialize(obj, apiKey, imapFqdn, imapPassword, imapUsername, name) {
      obj['api-key'] = apiKey;
      obj['imap-fqdn'] = imapFqdn;
      obj['imap-password'] = imapPassword;
      obj['imap-username'] = imapUsername;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateZeroSSLTarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateZeroSSLTarget} obj Optional instance to populate.
     * @return {module:model/UpdateZeroSSLTarget} The populated <code>UpdateZeroSSLTarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateZeroSSLTarget();

        if (data.hasOwnProperty('api-key')) {
          obj['api-key'] = _ApiClient["default"].convertToType(data['api-key'], 'String');
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('imap-fqdn')) {
          obj['imap-fqdn'] = _ApiClient["default"].convertToType(data['imap-fqdn'], 'String');
        }

        if (data.hasOwnProperty('imap-password')) {
          obj['imap-password'] = _ApiClient["default"].convertToType(data['imap-password'], 'String');
        }

        if (data.hasOwnProperty('imap-port')) {
          obj['imap-port'] = _ApiClient["default"].convertToType(data['imap-port'], 'String');
        }

        if (data.hasOwnProperty('imap-target-email')) {
          obj['imap-target-email'] = _ApiClient["default"].convertToType(data['imap-target-email'], 'String');
        }

        if (data.hasOwnProperty('imap-username')) {
          obj['imap-username'] = _ApiClient["default"].convertToType(data['imap-username'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('timeout')) {
          obj['timeout'] = _ApiClient["default"].convertToType(data['timeout'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('update-version')) {
          obj['update-version'] = _ApiClient["default"].convertToType(data['update-version'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return UpdateZeroSSLTarget;
}();
/**
 * API Key of the ZeroSSLTarget account
 * @member {String} api-key
 */


UpdateZeroSSLTarget.prototype['api-key'] = undefined;
/**
 * Deprecated - use description
 * @member {String} comment
 */

UpdateZeroSSLTarget.prototype['comment'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

UpdateZeroSSLTarget.prototype['description'] = undefined;
/**
 * ImapFQDN of the IMAP service, FQDN or IPv4 address. Must be FQDN if the IMAP is using TLS
 * @member {String} imap-fqdn
 */

UpdateZeroSSLTarget.prototype['imap-fqdn'] = undefined;
/**
 * ImapPassword to access the IMAP service
 * @member {String} imap-password
 */

UpdateZeroSSLTarget.prototype['imap-password'] = undefined;
/**
 * ImapPort of the IMAP service
 * @member {String} imap-port
 * @default '993'
 */

UpdateZeroSSLTarget.prototype['imap-port'] = '993';
/**
 * ImapValidationEmail to use when asking ZeroSSL to send a validation email, if empty will user imap-username
 * @member {String} imap-target-email
 */

UpdateZeroSSLTarget.prototype['imap-target-email'] = undefined;
/**
 * ImapUsername to access the IMAP service
 * @member {String} imap-username
 */

UpdateZeroSSLTarget.prototype['imap-username'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

UpdateZeroSSLTarget.prototype['json'] = false;
/**
 * Whether to keep previous version [true/false]. If not set, use default according to account settings
 * @member {String} keep-prev-version
 */

UpdateZeroSSLTarget.prototype['keep-prev-version'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

UpdateZeroSSLTarget.prototype['key'] = undefined;
/**
 * Target name
 * @member {String} name
 */

UpdateZeroSSLTarget.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

UpdateZeroSSLTarget.prototype['new-name'] = undefined;
/**
 * Timeout waiting for certificate validation in Duration format (1h - 1 Hour, 20m - 20 Minutes, 33m3s - 33 Minutes and 3 Seconds), maximum 1h.
 * @member {String} timeout
 * @default '5m'
 */

UpdateZeroSSLTarget.prototype['timeout'] = '5m';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateZeroSSLTarget.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateZeroSSLTarget.prototype['uid-token'] = undefined;
/**
 * Deprecated
 * @member {Boolean} update-version
 */

UpdateZeroSSLTarget.prototype['update-version'] = undefined;
var _default = UpdateZeroSSLTarget;
exports["default"] = _default;
