// akeyless/dist/model/UpdateGroup.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateGroup model module.
 * @module model/UpdateGroup
 * @version 3.5.0
 */
var UpdateGroup = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateGroup</code>.
   * @alias module:model/UpdateGroup
   * @param groupAlias {String} A short group alias
   * @param name {String} Group name
   */
  function UpdateGroup(groupAlias, name) {
    _classCallCheck(this, UpdateGroup);

    UpdateGroup.initialize(this, groupAlias, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateGroup, null, [{
    key: "initialize",
    value: function initialize(obj, groupAlias, name) {
      obj['group-alias'] = groupAlias;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateGroup</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateGroup} obj Optional instance to populate.
     * @return {module:model/UpdateGroup} The populated <code>UpdateGroup</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateGroup();

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('group-alias')) {
          obj['group-alias'] = _ApiClient["default"].convertToType(data['group-alias'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-assignment')) {
          obj['user-assignment'] = _ApiClient["default"].convertToType(data['user-assignment'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateGroup;
}();
/**
 * Description of the object
 * @member {String} description
 */


UpdateGroup.prototype['description'] = undefined;
/**
 * A short group alias
 * @member {String} group-alias
 */

UpdateGroup.prototype['group-alias'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

UpdateGroup.prototype['json'] = false;
/**
 * Group name
 * @member {String} name
 */

UpdateGroup.prototype['name'] = undefined;
/**
 * Group new name
 * @member {String} new-name
 */

UpdateGroup.prototype['new-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateGroup.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateGroup.prototype['uid-token'] = undefined;
/**
 * A json string defining the access permission assignment for this client
 * @member {String} user-assignment
 */

UpdateGroup.prototype['user-assignment'] = undefined;
var _default = UpdateGroup;
exports["default"] = _default;
