// akeyless/dist/model/UpdateGlobalSignAtlasTarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateGlobalSignAtlasTarget model module.
 * @module model/UpdateGlobalSignAtlasTarget
 * @version 3.5.0
 */
var UpdateGlobalSignAtlasTarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateGlobalSignAtlasTarget</code>.
   * @alias module:model/UpdateGlobalSignAtlasTarget
   * @param apiKey {String} API Key of the GlobalSign Atlas account
   * @param apiSecret {String} API Secret of the GlobalSign Atlas account
   * @param name {String} Target name
   */
  function UpdateGlobalSignAtlasTarget(apiKey, apiSecret, name) {
    _classCallCheck(this, UpdateGlobalSignAtlasTarget);

    UpdateGlobalSignAtlasTarget.initialize(this, apiKey, apiSecret, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateGlobalSignAtlasTarget, null, [{
    key: "initialize",
    value: function initialize(obj, apiKey, apiSecret, name) {
      obj['api-key'] = apiKey;
      obj['api-secret'] = apiSecret;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateGlobalSignAtlasTarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateGlobalSignAtlasTarget} obj Optional instance to populate.
     * @return {module:model/UpdateGlobalSignAtlasTarget} The populated <code>UpdateGlobalSignAtlasTarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateGlobalSignAtlasTarget();

        if (data.hasOwnProperty('api-key')) {
          obj['api-key'] = _ApiClient["default"].convertToType(data['api-key'], 'String');
        }

        if (data.hasOwnProperty('api-secret')) {
          obj['api-secret'] = _ApiClient["default"].convertToType(data['api-secret'], 'String');
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('mtls-cert-data-base64')) {
          obj['mtls-cert-data-base64'] = _ApiClient["default"].convertToType(data['mtls-cert-data-base64'], 'String');
        }

        if (data.hasOwnProperty('mtls-key-data-base64')) {
          obj['mtls-key-data-base64'] = _ApiClient["default"].convertToType(data['mtls-key-data-base64'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('timeout')) {
          obj['timeout'] = _ApiClient["default"].convertToType(data['timeout'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('update-version')) {
          obj['update-version'] = _ApiClient["default"].convertToType(data['update-version'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return UpdateGlobalSignAtlasTarget;
}();
/**
 * API Key of the GlobalSign Atlas account
 * @member {String} api-key
 */


UpdateGlobalSignAtlasTarget.prototype['api-key'] = undefined;
/**
 * API Secret of the GlobalSign Atlas account
 * @member {String} api-secret
 */

UpdateGlobalSignAtlasTarget.prototype['api-secret'] = undefined;
/**
 * Deprecated - use description
 * @member {String} comment
 */

UpdateGlobalSignAtlasTarget.prototype['comment'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

UpdateGlobalSignAtlasTarget.prototype['description'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

UpdateGlobalSignAtlasTarget.prototype['json'] = false;
/**
 * Whether to keep previous version [true/false]. If not set, use default according to account settings
 * @member {String} keep-prev-version
 */

UpdateGlobalSignAtlasTarget.prototype['keep-prev-version'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

UpdateGlobalSignAtlasTarget.prototype['key'] = undefined;
/**
 * Mutual TLS Certificate contents of the GlobalSign Atlas account encoded in base64, either mtls-cert-file-path or mtls-cert-data-base64 must be supplied
 * @member {String} mtls-cert-data-base64
 */

UpdateGlobalSignAtlasTarget.prototype['mtls-cert-data-base64'] = undefined;
/**
 * Mutual TLS Key contents of the GlobalSign Atlas account encoded in base64, either mtls-key-file-path or mtls-data-base64 must be supplied
 * @member {String} mtls-key-data-base64
 */

UpdateGlobalSignAtlasTarget.prototype['mtls-key-data-base64'] = undefined;
/**
 * Target name
 * @member {String} name
 */

UpdateGlobalSignAtlasTarget.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

UpdateGlobalSignAtlasTarget.prototype['new-name'] = undefined;
/**
 * Timeout waiting for certificate validation in Duration format (1h - 1 Hour, 20m - 20 Minutes, 33m3s - 33 Minutes and 3 Seconds), maximum 1h.
 * @member {String} timeout
 * @default '5m'
 */

UpdateGlobalSignAtlasTarget.prototype['timeout'] = '5m';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateGlobalSignAtlasTarget.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateGlobalSignAtlasTarget.prototype['uid-token'] = undefined;
/**
 * Deprecated
 * @member {Boolean} update-version
 */

UpdateGlobalSignAtlasTarget.prototype['update-version'] = undefined;
var _default = UpdateGlobalSignAtlasTarget;
exports["default"] = _default;
