// akeyless/dist/model/VenafiTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The VenafiTargetDetails model module.
 * @module model/VenafiTargetDetails
 * @version 3.5.0
 */
var VenafiTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>VenafiTargetDetails</code>.
   * @alias module:model/VenafiTargetDetails
   */
  function VenafiTargetDetails() {
    _classCallCheck(this, VenafiTargetDetails);

    VenafiTargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(VenafiTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>VenafiTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/VenafiTargetDetails} obj Optional instance to populate.
     * @return {module:model/VenafiTargetDetails} The populated <code>VenafiTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new VenafiTargetDetails();

        if (data.hasOwnProperty('venafi_api_key')) {
          obj['venafi_api_key'] = _ApiClient["default"].convertToType(data['venafi_api_key'], 'String');
        }

        if (data.hasOwnProperty('venafi_base_url')) {
          obj['venafi_base_url'] = _ApiClient["default"].convertToType(data['venafi_base_url'], 'String');
        }

        if (data.hasOwnProperty('venafi_tpp_access_token')) {
          obj['venafi_tpp_access_token'] = _ApiClient["default"].convertToType(data['venafi_tpp_access_token'], 'String');
        }

        if (data.hasOwnProperty('venafi_tpp_client_id')) {
          obj['venafi_tpp_client_id'] = _ApiClient["default"].convertToType(data['venafi_tpp_client_id'], 'String');
        }

        if (data.hasOwnProperty('venafi_tpp_password')) {
          obj['venafi_tpp_password'] = _ApiClient["default"].convertToType(data['venafi_tpp_password'], 'String');
        }

        if (data.hasOwnProperty('venafi_tpp_refresh_token')) {
          obj['venafi_tpp_refresh_token'] = _ApiClient["default"].convertToType(data['venafi_tpp_refresh_token'], 'String');
        }

        if (data.hasOwnProperty('venafi_tpp_username')) {
          obj['venafi_tpp_username'] = _ApiClient["default"].convertToType(data['venafi_tpp_username'], 'String');
        }

        if (data.hasOwnProperty('venafi_use_tpp')) {
          obj['venafi_use_tpp'] = _ApiClient["default"].convertToType(data['venafi_use_tpp'], 'Boolean');
        }

        if (data.hasOwnProperty('venafi_zone')) {
          obj['venafi_zone'] = _ApiClient["default"].convertToType(data['venafi_zone'], 'String');
        }
      }

      return obj;
    }
  }]);

  return VenafiTargetDetails;
}();
/**
 * @member {String} venafi_api_key
 */


VenafiTargetDetails.prototype['venafi_api_key'] = undefined;
/**
 * @member {String} venafi_base_url
 */

VenafiTargetDetails.prototype['venafi_base_url'] = undefined;
/**
 * @member {String} venafi_tpp_access_token
 */

VenafiTargetDetails.prototype['venafi_tpp_access_token'] = undefined;
/**
 * @member {String} venafi_tpp_client_id
 */

VenafiTargetDetails.prototype['venafi_tpp_client_id'] = undefined;
/**
 * Deprecated: VenafiAccessToken and VenafiRefreshToken should be used instead
 * @member {String} venafi_tpp_password
 */

VenafiTargetDetails.prototype['venafi_tpp_password'] = undefined;
/**
 * @member {String} venafi_tpp_refresh_token
 */

VenafiTargetDetails.prototype['venafi_tpp_refresh_token'] = undefined;
/**
 * Deprecated: VenafiAccessToken and VenafiRefreshToken should be used instead
 * @member {String} venafi_tpp_username
 */

VenafiTargetDetails.prototype['venafi_tpp_username'] = undefined;
/**
 * @member {Boolean} venafi_use_tpp
 */

VenafiTargetDetails.prototype['venafi_use_tpp'] = undefined;
/**
 * @member {String} venafi_zone
 */

VenafiTargetDetails.prototype['venafi_zone'] = undefined;
var _default = VenafiTargetDetails;
exports["default"] = _default;
