// akeyless/dist/model/VerifyRsaSsaPss.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The VerifyRsaSsaPss model module.
 * @module model/VerifyRsaSsaPss
 * @version 3.5.0
 */
var VerifyRsaSsaPss = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>VerifyRsaSsaPss</code>.
   * verifyRsaSsaPss is a command that Verifies an rsassa-pss signature
   * @alias module:model/VerifyRsaSsaPss
   * @param message {String} The input message to verify in a base64 format
   * @param signature {String} The message's signature
   */
  function VerifyRsaSsaPss(message, signature) {
    _classCallCheck(this, VerifyRsaSsaPss);

    VerifyRsaSsaPss.initialize(this, message, signature);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(VerifyRsaSsaPss, null, [{
    key: "initialize",
    value: function initialize(obj, message, signature) {
      obj['message'] = message;
      obj['signature'] = signature;
    }
    /**
     * Constructs a <code>VerifyRsaSsaPss</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/VerifyRsaSsaPss} obj Optional instance to populate.
     * @return {module:model/VerifyRsaSsaPss} The populated <code>VerifyRsaSsaPss</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new VerifyRsaSsaPss();

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('hash-function')) {
          obj['hash-function'] = _ApiClient["default"].convertToType(data['hash-function'], 'String');
        }

        if (data.hasOwnProperty('item-id')) {
          obj['item-id'] = _ApiClient["default"].convertToType(data['item-id'], 'Number');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key-name')) {
          obj['key-name'] = _ApiClient["default"].convertToType(data['key-name'], 'String');
        }

        if (data.hasOwnProperty('message')) {
          obj['message'] = _ApiClient["default"].convertToType(data['message'], 'String');
        }

        if (data.hasOwnProperty('signature')) {
          obj['signature'] = _ApiClient["default"].convertToType(data['signature'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return VerifyRsaSsaPss;
}();
/**
 * The display id of the RSA key to use in the verification process
 * @member {String} display-id
 */


VerifyRsaSsaPss.prototype['display-id'] = undefined;
/**
 * HashFunction defines the hash function (e.g. sha-256)
 * @member {String} hash-function
 */

VerifyRsaSsaPss.prototype['hash-function'] = undefined;
/**
 * The item id of the RSA key to use in the verification process
 * @member {Number} item-id
 */

VerifyRsaSsaPss.prototype['item-id'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

VerifyRsaSsaPss.prototype['json'] = false;
/**
 * The name of the RSA key to use in the verification process
 * @member {String} key-name
 */

VerifyRsaSsaPss.prototype['key-name'] = undefined;
/**
 * The input message to verify in a base64 format
 * @member {String} message
 */

VerifyRsaSsaPss.prototype['message'] = undefined;
/**
 * The message's signature
 * @member {String} signature
 */

VerifyRsaSsaPss.prototype['signature'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

VerifyRsaSsaPss.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

VerifyRsaSsaPss.prototype['uid-token'] = undefined;
var _default = VerifyRsaSsaPss;
exports["default"] = _default;
