// akeyless/dist/model/VerifyGPG.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The VerifyGPG model module.
 * @module model/VerifyGPG
 * @version 3.3.9
 */
var VerifyGPG = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>VerifyGPG</code>.
   * @alias module:model/VerifyGPG
   * @param keyName {String} The name of the key to use in the encryption process
   * @param signature {String} The signature to be verified in base64 format
   */
  function VerifyGPG(keyName, signature) {
    _classCallCheck(this, VerifyGPG);

    VerifyGPG.initialize(this, keyName, signature);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(VerifyGPG, null, [{
    key: "initialize",
    value: function initialize(obj, keyName, signature) {
      obj['key-name'] = keyName;
      obj['signature'] = signature;
    }
    /**
     * Constructs a <code>VerifyGPG</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/VerifyGPG} obj Optional instance to populate.
     * @return {module:model/VerifyGPG} The populated <code>VerifyGPG</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new VerifyGPG();

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('item-id')) {
          obj['item-id'] = _ApiClient["default"].convertToType(data['item-id'], 'Number');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key-name')) {
          obj['key-name'] = _ApiClient["default"].convertToType(data['key-name'], 'String');
        }

        if (data.hasOwnProperty('passphrase')) {
          obj['passphrase'] = _ApiClient["default"].convertToType(data['passphrase'], 'String');
        }

        if (data.hasOwnProperty('signature')) {
          obj['signature'] = _ApiClient["default"].convertToType(data['signature'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return VerifyGPG;
}();
/**
 * The display id of the key to use in the encryption process
 * @member {String} display-id
 */


VerifyGPG.prototype['display-id'] = undefined;
/**
 * The item id of the key to use in the encryption process
 * @member {Number} item-id
 */

VerifyGPG.prototype['item-id'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

VerifyGPG.prototype['json'] = false;
/**
 * The name of the key to use in the encryption process
 * @member {String} key-name
 */

VerifyGPG.prototype['key-name'] = undefined;
/**
 * Passphrase that was used to generate the key
 * @member {String} passphrase
 */

VerifyGPG.prototype['passphrase'] = undefined;
/**
 * The signature to be verified in base64 format
 * @member {String} signature
 */

VerifyGPG.prototype['signature'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

VerifyGPG.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

VerifyGPG.prototype['uid-token'] = undefined;
var _default = VerifyGPG;
exports["default"] = _default;
